<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">
    <link rel="icon" href="images/favicon.ico">

    <title>Enlight</title>

    <!-- Bootstrap core CSS -->
    <link href="css/bootstrap.min.css" rel="stylesheet">

    <!-- Custom styles for this template -->
    <link href="css/jumbotron.css" rel="stylesheet">

    <!-- IE10 viewport hack for Surface/desktop Windows 8 bug -->
    <script src="js/ie10-viewport-bug-workaround.js"></script>

    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
      <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->

    <script>
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
		(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','//www.google-analytics.com/analytics.js','ga');
	
	ga('create', 'UA-50646209-1', 'usc.edu');
	ga('send', 'pageview');
	
    </script>

  </head>

<body onload="changeTracks();toggle_single_multi_region($('#region_multi_single_button_id'));">

    <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
      <div class="container">
        <div class="navbar-header">
          <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
	  <a class="navbar-brand" href="index.html">Enlight</a>
        </div>
	<div class="navbar-collapse collapse">
	    <ul class="nav navbar-nav">
		<li class="active"><a href="index.html">Home</a></li>
		<li><a href="pages/tutorial.html">Tutorial</a></li>
		<li><a href="pages/help.html">Help</a></li>
		<li><a href="pages/pipeline.html">Pipeline</a></li>
		<li><a href="pages/faq.html">FAQ</a></li>
		<li class="dropdown">
		<a href="#" class="dropdown-toggle" data-toggle="dropdown">Example <span class="caret"></span></a>
		<ul class="dropdown-menu" role="menu">
		    <li><a href="pages/example/exampleinput.html">Example Input</a></li>
		    <li><a href="pages/example/exampleoutput/index.html">Example Output</a></li>
		</ul>
		</li>
	    </ul>
	</div><!--/.nav-collapse -->
    </div>
    </div>

    <!-- Main jumbotron for a primary marketing message or call to action -->
    <div class="jumbotron">
      <div class="container">
	  <h1>Enlight<br><small>integrating GWAS results with biological annotations</small></h1>
	  <p>Enlight draws regional plots for GWAS results, and overlays epigenetic modification, DNase sensitivity site, transcription factor binding annotation onto it. The combined plot will help identify causal variants. Users can also upload custom annotation, obtain text annotation for each SNP.
	  </p>
        <p><a class="btn btn-primary btn-lg" role="button" href="pages/help.html">Learn more &raquo;</a></p>
      </div>
    </div>

    <div class="container">
<script type="text/javascript"> //<![CDATA[

function changeTracks()
{
    var insertPos=document.getElementById('dataTrackHere');
    var all=[ {name:'wgEncodeBroadHistoneH1hescH3k27me3StdPk',cell:'H1-hESC',experiment:'ChIP-seq_H3K27me3'},
{name:'wgEncodeBroadHistoneHsmmH3k4me3StdPk',cell:'HSMM',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeAwgTfbsBroadHsmmCtcfUniPk',cell:'HSMM',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeBroadHistoneHelas3H3k09me3Pk',cell:'HeLa-S3',experiment:'ChIP-seq_H3K9me3'},
{name:'wgEncodeBroadHistoneHsmmH3k4me1StdPk',cell:'HSMM',experiment:'ChIP-seq_H3K4me1'},
{name:'wgEncodeUwTfbsMcf7CtcfStdPkRep1',cell:'MCF-7',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeBroadHistoneHepg2H3k9acStdPk',cell:'HepG2',experiment:'ChIP-seq_H3K9ac'},
{name:'wgEncodeBroadHistoneA549H3k79me2Etoh02Pk',cell:'A549',experiment:'ChIP-seq_H3K79me2'},
{name:'wgEncodeBroadHistoneNhekH3k36me3StdPk',cell:'NHEK',experiment:'ChIP-seq_H3K36me3'},
{name:'wgEncodeAwgTfbsBroadHmecCtcfUniPk',cell:'HMEC',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeBroadHistoneH1hescH3k4me3StdPk',cell:'H1-hESC',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeBroadHmmHepg2HMM',cell:'HepG2',experiment:'chromHMM'},
{name:'wgEncodeUwDnaseHelas3PkRep1',cell:'HeLa-S3',experiment:'DNase-seq'},
{name:'wgEncodeBroadHistoneHelas3H3k9acStdPk',cell:'HeLa-S3',experiment:'ChIP-seq_H3K9ac'},
{name:'wgEncodeHaibMethylRrbsHepg2DukeSitesRep1',cell:'HepG2',experiment:'methyl RRBS'},
{name:'wgEncodeBroadHistoneNhekH3k4me3StdPk',cell:'NHEK',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeBroadHistoneNhekH3k79me2Pk',cell:'NHEK',experiment:'ChIP-seq_H3K79me2'},
{name:'wgEncodeBroadHistoneHepg2H3k36me3StdPk',cell:'HepG2',experiment:'ChIP-seq_H3K36me3'},
{name:'wgEncodeAwgTfbsBroadGm12878CtcfUniPk',cell:'GM12878',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeSydhHistoneMcf7H3k36me3bUcdPk',cell:'MCF-7',experiment:'ChIP-seq_H3K36me3'},
{name:'wgEncodeBroadHistoneHuvecH3k36me3StdPk',cell:'HUVEC',experiment:'ChIP-seq_H3K36me3'},
{name:'wgEncodeBroadHistoneHepg2H3k27acStdPk',cell:'HepG2',experiment:'ChIP-seq_H3K27ac'},
{name:'wgEncodeBroadHistoneGm12878H3k9me3StdPk',cell:'GM12878',experiment:'ChIP-seq_H3K9me3'},
{name:'wgEncodeOpenChromFaireH1hescPk',cell:'H1-hESC',experiment:'FAIRE-seq'},
{name:'wgEncodeBroadHistoneHepg2H3k09me3Pk',cell:'HepG2',experiment:'ChIP-seq_H3K9me3'},
{name:'wgEncodeSydhHistoneMcf7H3k09me3UcdPk',cell:'MCF-7',experiment:'ChIP-seq_H3K9me3'},
{name:'wgEncodeBroadHistoneHelas3H3k04me1StdPk',cell:'HeLa-S3',experiment:'ChIP-seq_H3K4me1'},
{name:'wgEncodeBroadHistoneHepg2H3k27me3StdPk',cell:'HepG2',experiment:'ChIP-seq_H3K27me3'},
{name:'wgEncodeBroadHistoneGm12878H3k36me3StdPk',cell:'GM12878',experiment:'ChIP-seq_H3K36me3'},
{name:'wgEncodeSydhTfbsHepg2Tcf7l2UcdPk',cell:'HepG2',experiment:'ChIP-seq_TCF7L2'},
{name:'wgEncodeUwDnaseNhekPkRep1',cell:'NHEK',experiment:'DNase-seq'},
{name:'wgEncodeOpenChromFaireHuvecPk',cell:'HUVEC',experiment:'FAIRE-seq'},
{name:'wgEncodeUwHistoneMcf7H3k4me3StdPkRep1',cell:'MCF-7',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeOpenChromFaireHepg2Pk',cell:'HepG2',experiment:'FAIRE-seq'},
{name:'wgEncodeHaibMethylRrbsH1hescHaibSitesRep1',cell:'H1-hESC',experiment:'methyl RRBS'},
{name:'wgEncodeUwDnaseMcf7PkRep1',cell:'MCF-7',experiment:'DNase-seq'},
{name:'wgEncodeOpenChromChipHuvecCtcfPk',cell:'HUVEC',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeBroadHistoneNhlfH3k9acStdPk',cell:'NHLF',experiment:'ChIP-seq_H3K9ac'},
{name:'wgEncodeOpenChromDnaseA549Pk',cell:'A549',experiment:'DNase-seq'},
{name:'wgEncodeBroadHistoneHmecH3k36me3StdPk',cell:'HMEC',experiment:'ChIP-seq_H3K36me3'},
{name:'wgEncodeOpenChromFaireA549Pk',cell:'A549',experiment:'FAIRE-seq'},
{name:'wgEncodeSydhTfbsHelas3Tcf7l2UcdPk',cell:'HeLa-S3',experiment:'ChIP-seq_TCF7L2'},
{name:'wgEncodeBroadHistoneA549H3k27me3Etoh02Pk',cell:'A549',experiment:'ChIP-seq_H3K27me3'},
{name:'wgEncodeHaibMethylRrbsMcf7DukeSitesRep1',cell:'MCF-7',experiment:'methyl RRBS'},
{name:'wgEncodeBroadHistoneHuvecH3k4me3StdPk',cell:'HUVEC',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeHaibMethylRrbsHelas3HaibSitesRep1',cell:'HeLa-S3',experiment:'methyl RRBS'},
{name:'wgEncodeBroadHistoneHepg2H3k4me3StdPk',cell:'HepG2',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeOpenChromFaireGm12878Pk',cell:'GM12878',experiment:'FAIRE-seq'},
{name:'wgEncodeBroadHistoneHelas3H3k27acStdPk',cell:'HeLa-S3',experiment:'ChIP-seq_H3K27ac'},
{name:'wgEncodeBroadHistoneA549H3k09me3Etoh02Pk',cell:'A549',experiment:'ChIP-seq_H3K9me3'},
{name:'wgEncodeBroadHistoneA549H3k27acEtoh02Pk',cell:'A549',experiment:'ChIP-seq_H3K27ac'},
{name:'wgEncodeOpenChromDnaseHuvecPk',cell:'HUVEC',experiment:'DNase-seq'},
{name:'wgEncodeBroadHistoneGm12878H3k27acStdPk',cell:'GM12878',experiment:'ChIP-seq_H3K27ac'},
{name:'wgEncodeBroadHistoneNhlfH3k79me2Pk',cell:'NHLF',experiment:'ChIP-seq_H3K79me2'},
{name:'wgEncodeAwgTfbsHaibGm12878Tcf3Pcr1xUniPk',cell:'GM12878',experiment:'ChIP-seq_TCF3'},
{name:'wgEncodeUwDnaseNhlfPkRep1',cell:'NHLF',experiment:'DNase-seq'},
{name:'wgEncodeBroadHistoneHsmmH3k27me3StdPk',cell:'HSMM',experiment:'ChIP-seq_H3K27me3'},
{name:'wgEncodeBroadHistoneNhlfH3k4me1StdPk',cell:'NHLF',experiment:'ChIP-seq_H3K4me1'},
{name:'wgEncodeUwDnaseK562PkRep1',cell:'K562',experiment:'DNase-seq'},
{name:'wgEncodeBroadHistoneGm12878H3k27me3StdPk',cell:'GM12878',experiment:'ChIP-seq_H3K27me3'},
{name:'wgEncodeBroadHistoneNhekH3k09me3Pk',cell:'NHEK',experiment:'ChIP-seq_H3K9me3'},
{name:'wgEncodeBroadHistoneHelas3H3k79me2StdPk',cell:'HeLa-S3',experiment:'ChIP-seq_H3K79me2'},
{name:'wgEncodeUwTfbsCaco2CtcfStdPkRep1',cell:'Caco-2',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeBroadHistoneNhlfH3k36me3StdPk',cell:'NHLF',experiment:'ChIP-seq_H3K36me3'},
{name:'wgEncodeBroadHistoneNhlfH3k09me3Pk',cell:'NHLF',experiment:'ChIP-seq_H3K9me3'},
{name:'wgEncodeBroadHistoneGm12878H3k79me2StdPk',cell:'GM12878',experiment:'ChIP-seq_H3K79me2'},
{name:'wgEncodeUwHistoneCaco2H3k4me3StdPkRep1',cell:'Caco-2',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeBroadHistoneHuvecH3k27me3StdPk',cell:'HUVEC',experiment:'ChIP-seq_H3K27me3'},
{name:'wgEncodeBroadHmmK562HMM',cell:'K562',experiment:'chromHMM'},
{name:'wgEncodeBroadHistoneK562H3k4me3StdPk',cell:'K562',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeBroadHistoneHepg2H3k79me2StdPk',cell:'HepG2',experiment:'ChIP-seq_H3K79me2'},
{name:'wgEncodeSydhHistoneMcf7H3k27acUcdPk',cell:'MCF-7',experiment:'ChIP-seq_H3K27ac'},
{name:'wgEncodeBroadHistoneHelas3H3k27me3StdPk',cell:'HeLa-S3',experiment:'ChIP-seq_H3K27me3'},
{name:'wgEncodeBroadHistoneNhlfH3k4me3StdPk',cell:'NHLF',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeUwTfbsK562CtcfStdPkRep1',cell:'K562',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeBroadHistoneHmecH3k27acStdPk',cell:'HMEC',experiment:'ChIP-seq_H3K27ac'},
{name:'wgEncodeSydhHistoneMcf7H3k27me3bUcdPk',cell:'MCF-7',experiment:'ChIP-seq_H3K27me3'},
{name:'wgEncodeBroadHistoneGm12878H3k4me1StdPk',cell:'GM12878',experiment:'ChIP-seq_H3K4me1'},
{name:'wgEncodeOpenChromDnaseGm12878Pk',cell:'GM12878',experiment:'DNase-seq'},
{name:'wgEncodeAwgTfbsBroadNhekCtcfUniPk',cell:'NHEK',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeBroadHistoneHsmmH3k27acStdPk',cell:'HSMM',experiment:'ChIP-seq_H3K27ac'},
{name:'wgEncodeOpenChromFaireNhekPk',cell:'NHEK',experiment:'FAIRE-seq'},
{name:'wgEncodeBroadHistoneHsmmH3k36me3StdPk',cell:'HSMM',experiment:'ChIP-seq_H3K36me3'},
{name:'wgEncodeBroadHistoneHuvecH3k79me2Pk',cell:'HUVEC',experiment:'ChIP-seq_H3K79me2'},
{name:'wgEncodeBroadHistoneGm12878H3k9acStdPk',cell:'GM12878',experiment:'ChIP-seq_H3K9ac'},
{name:'wgEncodeRegTfbsClusteredV3',cell:'None',experiment:'TFBS Region'},
{name:'wgEncodeBroadHistoneHsmmH3k79me2StdPk',cell:'HSMM',experiment:'ChIP-seq_H3K79me2'},
{name:'wgEncodeUwHistoneCaco2H3k36me3StdPkRep1',cell:'Caco-2',experiment:'ChIP-seq_H3K36me3'},
{name:'wgEncodeBroadHistoneHmecH3k9acStdPk',cell:'HMEC',experiment:'ChIP-seq_H3K9ac'},
{name:'wgEncodeBroadHistoneH1hescH3k9acStdPk',cell:'H1-hESC',experiment:'ChIP-seq_H3K9ac'},
{name:'wgEncodeUwHistoneHct116H3k4me3StdPkRep1',cell:'HCT116',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeBroadHistoneK562H3k9me3StdPk',cell:'K562',experiment:'ChIP-seq_H3K9me3'},
{name:'wgEncodeBroadHistoneH1hescH3k09me3StdPk',cell:'H1-hESC',experiment:'ChIP-seq_H3K9me3'},
{name:'wgEncodeBroadHistoneK562H3k4me1StdPk',cell:'K562',experiment:'ChIP-seq_H3K4me1'},
{name:'wgEncodeUwDnaseHmecPkRep1',cell:'HMEC',experiment:'DNase-seq'},
{name:'wgEncodeBroadHistoneHsmmH3k9me3StdPk',cell:'HSMM',experiment:'ChIP-seq_H3K9me3'},
{name:'wgEncodeBroadHistoneHelas3H3k36me3StdPk',cell:'HeLa-S3',experiment:'ChIP-seq_H3K36me3'},
{name:'wgEncodeBroadHmmHsmmHMM',cell:'HSMM',experiment:'chromHMM'},
{name:'wgEncodeBroadHmmGm12878HMM',cell:'GM12878',experiment:'chromHMM'},
{name:'wgEncodeBroadHistoneK562H3k27me3StdPk',cell:'K562',experiment:'ChIP-seq_H3K27me3'},
{name:'wgEncodeBroadHmmNhekHMM',cell:'NHEK',experiment:'chromHMM'},
{name:'wgEncodeAwgTfbsUwHelas3CtcfUniPk',cell:'HeLa-S3',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeBroadHistoneA549H3k09acEtoh02Pk',cell:'A549',experiment:'ChIP-seq_H3K9ac'},
{name:'wgEncodeHaibMethylRrbsA549Dm002p7dHaibSitesRep1',cell:'A549',experiment:'methyl RRBS'},
{name:'wgEncodeHaibMethylRrbsK562HaibSitesRep1',cell:'K562',experiment:'methyl RRBS'},
{name:'wgEncodeBroadHmmNhlfHMM',cell:'NHLF',experiment:'chromHMM'},
{name:'wgEncodeBroadHistoneHuvecH3k4me1StdPk',cell:'HUVEC',experiment:'ChIP-seq_H3K4me1'},
{name:'wgEncodeBroadHistoneHmecH3k4me3StdPk',cell:'HMEC',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeBroadHistoneK562H3k36me3StdPk',cell:'K562',experiment:'ChIP-seq_H3K36me3'},
{name:'wgEncodeHaibMethylRrbsHct116StanfordSitesRep1',cell:'HCT116',experiment:'methyl RRBS'},
{name:'wgEncodeBroadHistoneHsmmH3k9acStdPk',cell:'HSMM',experiment:'ChIP-seq_H3K9ac'},
{name:'wgEncodeBroadHistoneA549H3k04me3Etoh02Pk',cell:'A549',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeSydhHistoneHct116H3k04me1UcdPk',cell:'HCT116',experiment:'ChIP-seq_H3K4me1'},
{name:'wgEncodeBroadHistoneK562H3k79me2StdPk',cell:'K562',experiment:'ChIP-seq_H3K79me2'},
{name:'wgEncodeUwHistoneCaco2H3k27me3StdPkRep1',cell:'Caco-2',experiment:'ChIP-seq_H3K27me3'},
{name:'wgEncodeOpenChromDnaseH1hescPk',cell:'H1-hESC',experiment:'DNase-seq'},
{name:'wgEncodeHaibMethylRrbsGm12878HaibSitesRep1',cell:'GM12878',experiment:'methyl RRBS'},
{name:'wgEncodeOpenChromChipH1hescCtcfPk',cell:'H1-hESC',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeBroadHistoneK562H3k9acStdPk',cell:'K562',experiment:'ChIP-seq_H3K9ac'},
{name:'wgEncodeBroadHistoneHmecH3k79me2Pk',cell:'HMEC',experiment:'ChIP-seq_H3K79me2'},
{name:'wgEncodeBroadHistoneA549H3k36me3Etoh02Pk',cell:'A549',experiment:'ChIP-seq_H3K36me3'},
{name:'wgEncodeOpenChromFaireHelas3Pk',cell:'HeLa-S3',experiment:'FAIRE-seq'},
{name:'wgEncodeBroadHmmH1hescHMM',cell:'H1-hESC',experiment:'chromHMM'},
{name:'wgEncodeBroadHistoneH1hescH3k36me3StdPk',cell:'H1-hESC',experiment:'ChIP-seq_H3K36me3'},
{name:'wgEncodeHaibTfbsHct116CtcfcV0422111PkRep1',cell:'HCT116',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeBroadHistoneH1hescH3k79me2StdPk',cell:'H1-hESC',experiment:'ChIP-seq_H3K79me2'},
{name:'wgEncodeBroadHistoneHmecH3k27me3StdPk',cell:'HMEC',experiment:'ChIP-seq_H3K27me3'},
{name:'wgEncodeBroadHistoneK562H3k27acStdPk',cell:'K562',experiment:'ChIP-seq_H3K27ac'},
{name:'wgEncodeBroadHmmHmecHMM',cell:'HMEC',experiment:'chromHMM'},
{name:'wgEncodeBroadHistoneH1hescH3k4me1StdPk',cell:'H1-hESC',experiment:'ChIP-seq_H3K4me1'},
{name:'wgEncodeBroadHistoneHuvecH3k27acStdPk',cell:'HUVEC',experiment:'ChIP-seq_H3K27ac'},
{name:'wgEncodeOpenChromDnaseHepg2Pk',cell:'HepG2',experiment:'DNase-seq'},
{name:'wgEncodeUwTfbsA549CtcfStdPkRep1',cell:'A549',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeBroadHistoneGm12878H3k4me3StdPk',cell:'GM12878',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeBroadHistoneNhekH3k9acStdPk',cell:'NHEK',experiment:'ChIP-seq_H3K9ac'},
{name:'wgEncodeBroadHistoneNhekH3k27me3StdPk',cell:'NHEK',experiment:'ChIP-seq_H3K27me3'},
{name:'wgEncodeBroadHistoneHepg2H3k04me1StdPk',cell:'HepG2',experiment:'ChIP-seq_H3K4me1'},
{name:'wgEncodeBroadHistoneNhlfCtcfStdPk',cell:'NHLF',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeOpenChromChipHepg2CtcfPk',cell:'HepG2',experiment:'ChIP-seq_CTCF'},
{name:'wgEncodeBroadHistoneA549H3k04me1Etoh02Pk',cell:'A549',experiment:'ChIP-seq_H3K4me1'},
{name:'wgEncodeBroadHistoneNhekH3k4me1StdPk',cell:'NHEK',experiment:'ChIP-seq_H3K4me1'},
{name:'wgEncodeUwDnaseHsmmPkRep1',cell:'HSMM',experiment:'DNase-seq'},
{name:'wgEncodeBroadHistoneNhlfH3k27me3StdPk',cell:'NHLF',experiment:'ChIP-seq_H3K27me3'},
{name:'wgEncodeBroadHmmHuvecHMM',cell:'HUVEC',experiment:'chromHMM'},
{name:'wgEncodeUwDnaseCaco2PkRep1',cell:'Caco-2',experiment:'DNase-seq'},
{name:'wgEncodeBroadHistoneHmecH3k09me3Pk',cell:'HMEC',experiment:'ChIP-seq_H3K9me3'},
{name:'wgEncodeSydhTfbsHct116Tcf7l2UcdPk',cell:'HCT116',experiment:'ChIP-seq_TCF7L2'},
{name:'wgEncodeUwDnaseHct116PkRep1',cell:'HCT116',experiment:'DNase-seq'},
{name:'wgEncodeBroadHistoneHuvecH3k9acStdPk',cell:'HUVEC',experiment:'ChIP-seq_H3K9ac'},
{name:'wgEncodeHaibMethylRrbsCaco2UwSitesRep1',cell:'Caco-2',experiment:'methyl RRBS'},
{name:'wgEncodeBroadHistoneH1hescH3k27acStdPk',cell:'H1-hESC',experiment:'ChIP-seq_H3K27ac'},
{name:'tfbsConsSites',cell:'None',experiment:'TFBS Conservation'},
{name:'wgEncodeBroadHistoneHelas3H3k4me3StdPk',cell:'HeLa-S3',experiment:'ChIP-seq_H3K4me3'},
{name:'wgEncodeBroadHistoneNhlfH3k27acStdPk',cell:'NHLF',experiment:'ChIP-seq_H3K27ac'},
{name:'wgEncodeBroadHistoneNhekH3k27acStdPk',cell:'NHEK',experiment:'ChIP-seq_H3K27ac'},
{name:'wgEncodeBroadHistoneHmecH3k4me1StdPk',cell:'HMEC',experiment:'ChIP-seq_H3K4me1'},
{name:'wgEncodeSydhHistoneHct116H3k27acUcdPk',cell:'HCT116',experiment:'ChIP-seq_H3K27ac'},
{name:'wgEncodeBroadHistoneHuvecH3k09me3Pk',cell:'HUVEC',experiment:'ChIP-seq_H3K9me3'} ];

    var cell=document.getElementsByClassName('cell');
    var experiment=document.getElementsByClassName('experiment');

    var selectedCell=[];
    var selectedExperiment=[];
    var tracks=[];

    //get selected cell
    for (var i=0;i<cell.length;i++)
    {
	if (cell[i].checked)
	{
	    selectedCell.push(cell[i].value);
	}
    }

    //get selected assay
    for (var i=0;i<experiment.length;i++)
    {
	if (experiment[i].checked)
	{
	    selectedExperiment.push(experiment[i].value);
	}
    }

    //get selected tracks
    for (var i=0;i<all.length;i++)
    {
	if (selectedCell.indexOf(all[i].cell) != -1)
	{
	    if (selectedExperiment.indexOf(all[i].experiment) != -1)
	    {
		tracks.push(all[i].name);
	    }
	}
    }

    //remove all child nodes while saving custom uploads info
    while(insertPos.firstChild)
    {
	insertPos.removeChild(insertPos.firstChild);
    }

    //add selected tracks
    for (var i=0;i<tracks.length;i++)
    {
	if (i>19)
	{
	    alert('At most 20 tracks can be selected.');
	    break;
	}
	var newrow=document.createElement('tr');
	var col=document.createElement('td');
	var label=document.createElement('label');
	var checkbox=document.createElement('input');
	label.innerHTML=tracks[i];
	checkbox.type='checkbox';
	checkbox.value=tracks[i];
	checkbox.name='generic_table';
	checkbox.checked=true;

	label.appendChild(checkbox);
	col.appendChild(label);
	newrow.appendChild(col);
	insertPos.appendChild(newrow);
    }

    //add custom tracks upload fields
    for (var i=0;i<20-tracks.length;i++)
    {
	var newrow=document.createElement('tr');
	var col=document.createElement('td');
	var span=document.createElement('span');
	var label=document.createElement('label');
	var upload=document.createElement('input');
	span.title='Drag or use Ctrl, Shift to select multiple files';
	label.innerHTML='Custom track (BED format;.gz okay)';
	upload.type='file';
	upload.name='custom_table';
	upload.multiple='multiple';

	label.appendChild(upload);
	span.appendChild(label);
	col.appendChild(span);
	newrow.appendChild(col);
	insertPos.appendChild(newrow);
    }
}

function response_to_select_region(caller)
{
    var region_spec_container=$(caller).parentsUntil("table.single_region_spec_head").find("td.region_detail_area");
    if($(caller).val()=='snp')
    {
	$(region_spec_container).find('table.snp_region_spec').show();
	$(region_spec_container).find('table').not('.snp_region_spec').hide();
    }
    else if ($(caller).val()=='gene')
    {
	$(region_spec_container).find('table.gene_region_spec').show();
	$(region_spec_container).find('table').not('.gene_region_spec').hide();
    }
    else if($(caller).val()=='chr')
    {
	$(region_spec_container).find('table.chr_region_spec').show();
	$(region_spec_container).find('table').not('.chr_region_spec').hide();
    }
}

function response_to_multi_select_region(caller)
{
    var file_region_spec_container=$("#file_region_specification_div_id");
    var manual_region_spec_container=$("#multi_manual_region_specification_div_id");

    if($(caller).val()=='region_file')
    {
	$(file_region_spec_container).show();
	$(manual_region_spec_container).hide();
    }
    else if($(caller).val()=='multi_region')
    {
	$(manual_region_spec_container).show();
	$(file_region_spec_container).hide();
	$(manual_region_spec_container).find("table.single_region_spec_head").each(
		function()
		{
		  var i=$(this).find("td.region_method_area input").first()
                      i.trigger("click");
		      i.prop('checked',true);
		}
	);
    }
}

//display and hide single region selection and multi-region selection alternatively
function toggle_single_multi_region(caller)
{
    var single_container=$("#region_specification_div_id");
    var multi_container=$("#multi_region_specification_div_id");
    if($('#region_multi_single_hidden_id').val()=='single')
    {
        $('#region_multi_single_hidden_id').val('multi');
	$(multi_container).hide();
        $(single_container).show();
	$(single_container).find("td.region_method_area input").first().trigger("click");
    }
    else if ($('#region_multi_single_hidden_id').val()=='multi')
    {
        $('#region_multi_single_hidden_id').val('single');
	$(single_container).hide();
        $(multi_container).show();
	$(multi_container).find("td.multi_region_method_area input").first().trigger("click");
    }
}

function clear_datatrack_selection()
{
    var cell=document.getElementsByClassName( 'cell');
    for (var i=0;i<cell.length;i++)
    {
	cell[i].checked=false;
    }
    var experiment=document.getElementsByClassName( 'experiment');
    for (var i=0;i<experiment.length;i++)
    {
	experiment[i].checked=false;
    }
}
function loadExampleSetting()
{
    document.getElementById('qformat_whitespace').checked=true;
    document.getElementById('markercol_id').value='dbSNP135';
    document.getElementById('pvalcol_id').value='P';
    document.getElementById('varAnno_id').value='UChicago_eQTL';
    document.getElementById('source_ref_pop_id').value='1000G_March2012,hg19,EUR';	

    $("#region_multi_single_hidden_id").val("single");
    $("#region_multi_single_button_id").trigger("click");

    $("td.region_detail_area input[name='snpflank']").val("20");
    $("td.region_detail_area input[name='refsnp']").val("rs2071278");

    document.getElementById('generic_toggle_id').checked=true;
    document.getElementById('anno_toggle_id').checked=true;
    document.getElementById('avinput_id').checked=false;
    document.getElementById('ref_hg19').checked=true;

    //set advanced options
    document.getElementById('ld_toggle_id').checked=true;

    //set interaction options
    $('#interaction_cell_type_k562_id').prop('checked',true);
    $('#interaction_cell_type_gm06690_id').prop('checked',false);
    $('#interaction_type_intra_id').prop('checked',true);
    $('#interaction_type_intra_id').trigger('click');
    $('#interaction_type_inter_id').prop('checked',false);
    $('[name="heatmap_toggle"]').prop('checked',true);


    //clear data track selection
    clear_datatrack_selection();
    document.getElementById('HSMM').checked=true;	
    document.getElementById('K562').checked=true;	
    document.getElementById('HepG2').checked=true;	
    document.getElementById('GM12878').checked=true;	
    document.getElementById('None').checked=true;	

    document.getElementById('TFBS Conservation').checked=true;
    document.getElementById('chromHMM').checked=true;
    document.getElementById('TFBS Region').checked=true;
    changeTracks();

    alert('Example settings loaded.');
}
function loadExampleInput()
{

    $('#query_example_label_div_id').show();
    $('#query_hidden_id').val('1');
    $('#query_file_div_id').hide();

    alert('Example input loaded.');
} 
function hideDetail()
{
    document.getElementById('option_detail_id').style.display='none';
}
function showDetail()
{
    document.getElementById('option_detail_id').style.display='table';
}
function check_before_submission()
{
   if($('#query_hidden_id').val() == '0')
   {//we're NOT using example input
    var query_file=$('#query_file_id').val();
    var query_url=$('#query_URL_id').val();

    if (query_file.length>0 && query_url.length>0)
    {
	alert('Input URL cannot be used with input file!');
	//abort submission
	return false;
    }

    if (query_file.length==0 && query_url.length==0)
    {
	alert('No input!');
	return false;
    }
   }

    //check email

    var email=document.getElementById('email_id').value;
    if (email)
    {
    	var email_pattern=/[\w\-\.]+\@[\w\-\.]+\.[\w\-\.]+/;

    	if (! email_pattern.test(email))
    	{
    	    alert('Illegal email address!');
    	    return false;
    	}
    } else
    {
    //when multiple regions are specified, the job will take a while
    //user should provide email
        if($("#region_multi_single_hidden_id").val()=='single')
        {
            alert('Email must be provided when multiple regions are to be plotted');
            return false;
        }
    }

    //check genome build
    var ref=document.getElementById('ref_hg19').checked;
    var ref_ld=document.getElementById('source_ref_pop_id').value.split(',');
    ref_ld=ref_ld[1];

    if (ref)
    {
	ref='hg19';
    } else
    {
	ref='hg18';
    }

    if (ref != ref_ld)
    {
	alert('Reference genome does not match source population');
	return false;
    }

    //check marker column
    var marker=document.getElementById('markercol_id').value;

    if (marker.length==0)
    {
	alert('Marker column name is empty!');
	return false;
    }

    //check P value column
    var p_val=document.getElementById('pvalcol_id').value;

    if (p_val.length==0)
    {
	alert('P value column name is empty!');
	return false;
    }

    //Only letters, numbers, dashes, underscores are allowed in column name
    var col_pat=/[^\w\-]/;

    if (col_pat.test(p_val) || col_pat.test(marker))
    {
	alert('Only letters, numbers, dashes, underscores are allowed in column name.');
	return false;
    }

    //check datatracks
    var generic_toggle=document.getElementById('generic_toggle_id').checked;
    var anno_toggle=document.getElementById('anno_toggle_id').checked;
    var datatrack=$('[name="generic_table"]').filter(':checked');
    var custom_table=document.getElementsByName('custom_table');
    var custom_table_count=0;
    $(custom_table).each(function(){
		    custom_table_count+=this.files.length;
		    });

    if ( generic_toggle || anno_toggle )
    {
	if (datatrack.length==0 && custom_table_count==0)
	{
	    alert ("No annotation data tracks selected or uploaded \nwhile generic plot or annotation is enabled.");
	    return false;
	}
    }
    if ( datatrack.length+custom_table_count>20)
    {
	alert('Too many data tracks (max: 20)');
	return false;
    }


    //region specification
    var region_pat=/[ \$\t\r\n\*\|\?\>\<\'\"\,\;\:\[\]\{\}]/;
    var region_method;
    var return_value=true;
    var at_least_one=false;
    
    //when toggled to multi view, the value of toggler is single and vice versa
    if ($('#region_multi_single_hidden_id').val()=='single')
    {
    	region_method=$('#multi_region_specification_div_id').find("input[name^='region_method']:checked");//select input elements with name starting with region_method
    }
    else if ($('#region_multi_single_hidden_id').val()=='multi')
    {
    	region_method=$('#region_specification_div_id').find("input[name^='region_method']:checked");//select input elements with name starting with region_method
    }
    
    //user must either upload a region specification file
    //or specify region for each region specification table

	if(! $("input[name='region_file']").val() )
	{
		$(region_method).each(
				function() {
				var region_spec_container=$(this).parentsUntil("table.single_region_spec_head").find("td.region_detail_area");
				var refsnp=$(region_spec_container).find("input[name^='refsnp']").val();
				var snpflank=$(region_spec_container).find("input[name^='snpflank']").val();
				var refgene=$(region_spec_container).find("input[name^='refgene']").val();
				var geneflank=$(region_spec_container).find("input[name^='geneflank']").val();
				var generefsnp=$(region_spec_container).find("input[name^='refsnp_for_gene']").val();
				var chr=$(region_spec_container).find("select[name^='chr']").val();
				var start=$(region_spec_container).find("input[name^='start']").val();
				var end=$(region_spec_container).find("input[name^='end']").val();
				var chrrefsnp=$(region_spec_container).find("input[name^='refsnp_for_chr']").val();

					if ($(this).val() == 'snp')
					{	
						if ( (refsnp.length!=0 && snpflank.length==0) || (refsnp.length==0 && snpflank.length!=0) )
						{
							alert('Both reference SNP and flanking region must be supplied.');
							return_value=false;
							return false;//stop each iteration
						} else if (region_pat.test(refsnp) || region_pat.test(snpflank))
						{
							alert("Illegal characters found in reference SNP or flanking regin\nPlease remove $ ' \" { } [ ] \\ > < : ; , * | and tab, space, newline.");
							return_value=false;
							return false;
						}	
						if( refsnp.length!=0 && snpflank.length!=0)
						{
							at_least_one=true;
						}
					} else if ($(this).val() == 'gene')
					{
						if ( (refgene.length!=0 && geneflank.length==0) || (refgene.length==0 && geneflank.length!=0) )
						{
							alert('Both reference gene and flanking region must be supplied.');
							return_value=false;
							return false;
						} else if (region_pat.test(refgene) || region_pat.test(geneflank) || region_pat.test(generefsnp))
						{
							alert("Illegal characters found in reference SNP or flanking regin\nPlease remove $ ' \" { } [ ] \\ > < : ; , * | and tab, space, newline.");
							return_value=false;
							return false;
						}	
						if( refgene.length!=0 && geneflank.length!=0)
						{
							at_least_one=true;
						}
					} else if ($(this).val() == 'chr')
					{
						if ( (start.length!=0 && end.length==0) || (start.length==0 && end.length!=0) )
						{
							alert('Both chromosome start and end must be supplied');
							return_value=false;
							return false;
						} else if (region_pat.test(chr) || region_pat.test(start) || region_pat.test(end) || region_pat.test(chrrefsnp))
						{
							alert("Illegal characters found in reference SNP or flanking regin\nPlease remove $ ' \" { } [ ] \\ > < : ; , * | and tab, space, newline.");
							return_value=false;
							return false;
						}	
						if( start.length!=0 && end.length!=0)
						{
							at_least_one=true;
						}
					}
			}
		);
		if(!at_least_one && return_value)
		{
			//when return_value is false, an alert has been issued, no need to give more alerts
			alert('No region is specified!');
		}
		return return_value&&at_least_one;
	} else
	{
		return true;
	}
}

//]]></script>
<noscript>
	<h1>
		Your browser does not support JavaScript! </br>Please enable JavaScript to use Enlight.
	</h1>
</noscript>
<form method="post" class="form-horizontal" role="form" action="/cgi-bin/process.cgi" enctype="multipart/form-data" name="main" onsubmit="return check_before_submission();"><h2>
	Input
</h2>
				<div class="form-group">
				    <label class="col-sm-3 control-label">DEMO</label>
				    <div class="col-sm-7">
				<button class="btn btn-info" type='button' onclick='loadExampleSetting()'>Load settings for example input</button>
	    <button type='button' class="btn btn-info" onclick='loadExampleInput()'>Load example input</button>
	</div>
    </div>
	<div class="form-group">
	    <label class="col-sm-3 control-label"><span title='receive result link'>Email (mandatory for multi-region)</span></label>
	    <div class="col-sm-7">
		<input name='email' id='email_id' type='email' placeholder="Enter Email" class="form-control"/>
	    </div>
	</div>
	<div class="form-group">
	    <label class="col-sm-3 control-label">Input file (1st line is header; .gz okay) </label>
	    <div class="col-sm-7">
		<div id="query_file_div_id">
		    <input type="file" name="query"  id="query_file_id" class="form-control"/> <strong >OR</strong>
		    <input name='query_URL' class="form-control" type='url' id='query_URL_id' placeholder="Paste URL here"/>
		</div> 
		<div id='query_example_label_div_id' style='display:none'>
		    <input class="form-control" id='query_hidden_id' type='hidden' name='example_upload' value=''/> 
		    <span id='query_example_label'>Example input loaded</span>
		</div>
	    </div>
	</div>
<div class="form-group">
    <label class="col-sm-3 control-label">Field delimiter</label>
    <div class="col-sm-7">
	<div class="radio">
	<label>
	 <input type="radio" name="qformat" value="whitespace" checked="checked" id="qformat_whitespace" />whitespace
</label> <label>
	 <input type="radio" name="qformat" value="tab" id="qformat_whitespace" />tab
</label> <label>
	 <input type="radio" name="qformat" value="space" id="qformat_whitespace" />space
</label> <label>
	 <input type="radio" name="qformat" value="comma" id="qformat_whitespace" />comma
</label>
</div>
</div>
</div>
<div class="form-group">
    <label class="col-sm-3 control-label"><span title='this columns contains rsID'>Marker Column (case sensitive)</span></label>
    <div class="col-sm-7">
	<input class="form-control" type="text" name="markercol"  id="markercol_id" placeholder="Enter marker column name"/>
</div>
</div>
<div class="form-group">
    <label class="col-sm-3 control-label">P value column (case-sensitive)</label>
    <div class="col-sm-7"><input type="text" class="form-control" name="pvalcol"  id="pvalcol_id" /></div>
</div>
<div class="form-group">
    <label class="col-sm-3 control-label">Genome Build</label>
    <div class="col-sm-7">
	    <label class="radio-inline"><input type="radio" name="ref" value="hg18">hg18</label>
	    <label class="radio-inline"><input type="radio" name="ref" id="ref_hg19" value="hg19" checked="checked">hg19</label>
	    </div>
	</div>
	    <div class="form-group">
		<label class="control-label col-sm-3">
		    <span title='choose data set for computing Linkage Disequilibrium'>Genome Build/LD source/Population</span> 
		</label>
		<div class="col-sm-7 ">
		    <select class="form-control" name="source_ref_pop"  id="source_ref_pop_id">
<option value="1000G_Aug2009,hg18,CEU">1000G_Aug2009,hg18,CEU</option>
<option value="1000G_Aug2009,hg18,JPT+CHB">1000G_Aug2009,hg18,JPT+CHB</option>
<option value="1000G_Aug2009,hg18,YRI">1000G_Aug2009,hg18,YRI</option>
<option value="1000G_June2010,hg18,CEU">1000G_June2010,hg18,CEU</option>
<option value="1000G_June2010,hg18,JPT+CHB">1000G_June2010,hg18,JPT+CHB</option>
<option value="1000G_June2010,hg18,YRI">1000G_June2010,hg18,YRI</option>
<option value="1000G_March2012,hg19,AFR">1000G_March2012,hg19,AFR</option>
<option value="1000G_March2012,hg19,AMR">1000G_March2012,hg19,AMR</option>
<option value="1000G_March2012,hg19,ASN">1000G_March2012,hg19,ASN</option>
<option selected="selected" value="1000G_March2012,hg19,EUR">1000G_March2012,hg19,EUR</option>
<option value="1000G_Nov2010,hg19,AFR">1000G_Nov2010,hg19,AFR</option>
<option value="1000G_Nov2010,hg19,ASN">1000G_Nov2010,hg19,ASN</option>
<option value="1000G_Nov2010,hg19,EUR">1000G_Nov2010,hg19,EUR</option>
<option value="hapmap,hg18,CEU">hapmap,hg18,CEU</option>
<option value="hapmap,hg18,JPT+CHB">hapmap,hg18,JPT+CHB</option>
<option value="hapmap,hg18,YRI">hapmap,hg18,YRI</option>
</select>
</div>
</div>
<div class="form-group">
    <label class="col-sm-3 control-label">
    <span title='show if a variant appears in a particular database'>Mark variants in database: </span> 
</label>
<div class="col-sm-7">
    <select name="varAnno" class="form-control" id="varAnno_id">
<option value="GTEX-140712.Brain_Cerebellum">GTEX-140712.Brain_Cerebellum</option>
<option value="GTEX-140712.Brain_Frontal_Cortex">GTEX-140712.Brain_Frontal_Cortex</option>
<option value="GTEX-140712.Brain_Pons">GTEX-140712.Brain_Pons</option>
<option value="GTEX-140712.Brain_Temporal_Cortex">GTEX-140712.Brain_Temporal_Cortex</option>
<option value="GTEX-140712.Liver">GTEX-140712.Liver</option>
<option value="GTEX-140712.Lymphoblastoid_RNAseq">GTEX-140712.Lymphoblastoid_RNAseq</option>
<option value="GTEX-140712.Lymphoblastoid_microarray">GTEX-140712.Lymphoblastoid_microarray</option>
<option value="GWAS_CATALOG_20140820">GWAS_CATALOG_20140820</option>
<option value="UChicago_eQTL">UChicago_eQTL</option>
<option selected="selected" value="NULL">NULL</option>
</select>
</div>
</div>
<hr>
<h2>
	<span title='plot region'>Specify a region</span>
</h2>
<div class="form-group">
    <label class="control-label col-sm-3">How many regions?</label>
    <div class="col-sm-7">
	    <label class="radio-inline"><input type="radio" name="region_multi_single_button" id="region_multi_single_button_multi_id" value="multi">multiple</label>
	    <label class="radio-inline"><input type="radio" name="region_multi_single_button" id="region_multi_single_button_single_id" value="single" checked="checked">single</label>
		    </div>
		    </div>
			<div id='region_specification_div_id' style='display:block'><table border=1 class='single_region_spec_head left_aln'>
				<div class='region_method_area'>
	    <input type='radio' name='region_method' onclick='response_to_select_region(this)' value='snp' checked >Reference SNP<br>
	     <input type='radio' name='region_method' onclick='response_to_select_region(this)' value='gene'>Reference Gene<br>
	     <input type='radio' name='region_method' onclick='response_to_select_region(this)' value='chr'>Chromosomal Region<br>
	    </div>
	    <div class='region_detail_area'>
		<div class='snp_region_spec'><tr>
			<div class="form-group">
			    <label class="col-sm-3 control-label">Index SNP</label>
			    <div class="col-sm-7">
					<input class="form-control" type='text' name='refsnp' size=15 />
				    </div>
				</div>
			
					<td>Flanking region (kb)</td>
					<td><input type='text' name='snpflank' size=15 value=''/></td>
					</tr></table> <table class='gene_region_spec'>
			                <tr>
					<td>Reference Gene</td>
					<td><input type='text' name='refgene' size=15 /></td>
					</tr>
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='geneflank' size=15 value='' /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_gene' size=15 /></td>
					</tr></table> <table class='chr_region_spec'>
					<tr>
					<td>Chromosome</td>
					<td><select name='chr' ><option value='1'>1</option>
<option value='2'>2</option>
<option value='3'>3</option>
<option value='4'>4</option>
<option value='5'>5</option>
<option value='6'>6</option>
<option value='7'>7</option>
<option value='8'>8</option>
<option value='9'>9</option>
<option value='10'>10</option>
<option value='11'>11</option>
<option value='12'>12</option>
<option value='13'>13</option>
<option value='14'>14</option>
<option value='15'>15</option>
<option value='16'>16</option>
<option value='17'>17</option>
<option value='18'>18</option>
<option value='19'>19</option>
<option value='20'>20</option>
<option value='21'>21</option>
<option value='22'>22</option>
<option value='X'>X</option></select></td>
					</tr>
					<tr>
					<td>Start (Mb)</td>
					<td><input type='text' name='start' size=15 /></td>
					<tr>
					<td>End (Mb)</td>
					<td><input type='text' name='end' size=15 /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_chr' size=15 /></td>
					</tr>
			</table> </td>
			</tr></table></div><div style='display:none' id='multi_region_specification_div_id'> <table class='noborder'>
			<tr>
			<td class='multi_region_method_area'>
			<input type='radio' name='multi_region_method' onclick='response_to_multi_select_region(this)' value='multi_region'>Manually Specify multi-region<br>
			<input type='radio' name='multi_region_method' onclick='response_to_multi_select_region(this)' value='region_file'>Region Spefication File<br>
			<tr>
			<td class='multi_region_detail_area'><div id='file_region_specification_div_id'><table class='noborder'>
					<tr>
					<td colspan=2>HitSpec format is supported (<a href='http://genome.sph.umich.edu/wiki/LocusZoom#Generating_a_Hit_Spec_File'>help</a>), <br>BUT 7th column and beyond are IGNORED
			</td>
			<td>
		<input type='file' name='region_file' size=15 />
</td> </tr>
			</table></div> <div id='multi_manual_region_specification_div_id'><table border=1><tr><td><table border=1 class='single_region_spec_head left_aln'>
			<tr>
				<td class='region_method_area'>
	    <input type='radio' name='region_method0' onclick='response_to_select_region(this)' value='snp' checked >Reference SNP<br>
	     <input type='radio' name='region_method0' onclick='response_to_select_region(this)' value='gene'>Reference Gene<br>
	     <input type='radio' name='region_method0' onclick='response_to_select_region(this)' value='chr'>Chromosomal Region<br>
	    </td>
			</tr>
			<tr>
			<td class='region_detail_area'><table class='snp_region_spec'><tr>
					<td>Index SNP</td>
					<td><input type='text' name='refsnp0' size=15 /></td>
					</tr>
			
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='snpflank0' size=15 value=''/></td>
					</tr></table> <table class='gene_region_spec'>
			                <tr>
					<td>Reference Gene</td>
					<td><input type='text' name='refgene0' size=15 /></td>
					</tr>
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='geneflank0' size=15 value='' /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_gene0' size=15 /></td>
					</tr></table> <table class='chr_region_spec'>
					<tr>
					<td>Chromosome</td>
					<td><select name='chr0' ><option value='1'>1</option>
<option value='2'>2</option>
<option value='3'>3</option>
<option value='4'>4</option>
<option value='5'>5</option>
<option value='6'>6</option>
<option value='7'>7</option>
<option value='8'>8</option>
<option value='9'>9</option>
<option value='10'>10</option>
<option value='11'>11</option>
<option value='12'>12</option>
<option value='13'>13</option>
<option value='14'>14</option>
<option value='15'>15</option>
<option value='16'>16</option>
<option value='17'>17</option>
<option value='18'>18</option>
<option value='19'>19</option>
<option value='20'>20</option>
<option value='21'>21</option>
<option value='22'>22</option>
<option value='X'>X</option></select></td>
					</tr>
					<tr>
					<td>Start (Mb)</td>
					<td><input type='text' name='start0' size=15 /></td>
					<tr>
					<td>End (Mb)</td>
					<td><input type='text' name='end0' size=15 /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_chr0' size=15 /></td>
					</tr>
			</table> </td>
			</tr></table></td><td><table border=1 class='single_region_spec_head left_aln'>
			<tr>
				<td class='region_method_area'>
	    <input type='radio' name='region_method1' onclick='response_to_select_region(this)' value='snp' checked >Reference SNP<br>
	     <input type='radio' name='region_method1' onclick='response_to_select_region(this)' value='gene'>Reference Gene<br>
	     <input type='radio' name='region_method1' onclick='response_to_select_region(this)' value='chr'>Chromosomal Region<br>
	    </td>
			</tr>
			<tr>
			<td class='region_detail_area'><table class='snp_region_spec'><tr>
					<td>Index SNP</td>
					<td><input type='text' name='refsnp1' size=15 /></td>
					</tr>
			
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='snpflank1' size=15 value=''/></td>
					</tr></table> <table class='gene_region_spec'>
			                <tr>
					<td>Reference Gene</td>
					<td><input type='text' name='refgene1' size=15 /></td>
					</tr>
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='geneflank1' size=15 value='' /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_gene1' size=15 /></td>
					</tr></table> <table class='chr_region_spec'>
					<tr>
					<td>Chromosome</td>
					<td><select name='chr1' ><option value='1'>1</option>
<option value='2'>2</option>
<option value='3'>3</option>
<option value='4'>4</option>
<option value='5'>5</option>
<option value='6'>6</option>
<option value='7'>7</option>
<option value='8'>8</option>
<option value='9'>9</option>
<option value='10'>10</option>
<option value='11'>11</option>
<option value='12'>12</option>
<option value='13'>13</option>
<option value='14'>14</option>
<option value='15'>15</option>
<option value='16'>16</option>
<option value='17'>17</option>
<option value='18'>18</option>
<option value='19'>19</option>
<option value='20'>20</option>
<option value='21'>21</option>
<option value='22'>22</option>
<option value='X'>X</option></select></td>
					</tr>
					<tr>
					<td>Start (Mb)</td>
					<td><input type='text' name='start1' size=15 /></td>
					<tr>
					<td>End (Mb)</td>
					<td><input type='text' name='end1' size=15 /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_chr1' size=15 /></td>
					</tr>
			</table> </td>
			</tr></table></td><td><table border=1 class='single_region_spec_head left_aln'>
			<tr>
				<td class='region_method_area'>
	    <input type='radio' name='region_method2' onclick='response_to_select_region(this)' value='snp' checked >Reference SNP<br>
	     <input type='radio' name='region_method2' onclick='response_to_select_region(this)' value='gene'>Reference Gene<br>
	     <input type='radio' name='region_method2' onclick='response_to_select_region(this)' value='chr'>Chromosomal Region<br>
	    </td>
			</tr>
			<tr>
			<td class='region_detail_area'><table class='snp_region_spec'><tr>
					<td>Index SNP</td>
					<td><input type='text' name='refsnp2' size=15 /></td>
					</tr>
			
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='snpflank2' size=15 value=''/></td>
					</tr></table> <table class='gene_region_spec'>
			                <tr>
					<td>Reference Gene</td>
					<td><input type='text' name='refgene2' size=15 /></td>
					</tr>
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='geneflank2' size=15 value='' /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_gene2' size=15 /></td>
					</tr></table> <table class='chr_region_spec'>
					<tr>
					<td>Chromosome</td>
					<td><select name='chr2' ><option value='1'>1</option>
<option value='2'>2</option>
<option value='3'>3</option>
<option value='4'>4</option>
<option value='5'>5</option>
<option value='6'>6</option>
<option value='7'>7</option>
<option value='8'>8</option>
<option value='9'>9</option>
<option value='10'>10</option>
<option value='11'>11</option>
<option value='12'>12</option>
<option value='13'>13</option>
<option value='14'>14</option>
<option value='15'>15</option>
<option value='16'>16</option>
<option value='17'>17</option>
<option value='18'>18</option>
<option value='19'>19</option>
<option value='20'>20</option>
<option value='21'>21</option>
<option value='22'>22</option>
<option value='X'>X</option></select></td>
					</tr>
					<tr>
					<td>Start (Mb)</td>
					<td><input type='text' name='start2' size=15 /></td>
					<tr>
					<td>End (Mb)</td>
					<td><input type='text' name='end2' size=15 /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_chr2' size=15 /></td>
					</tr>
			</table> </td>
			</tr></table></td></tr><tr><td><table border=1 class='single_region_spec_head left_aln'>
			<tr>
				<td class='region_method_area'>
	    <input type='radio' name='region_method3' onclick='response_to_select_region(this)' value='snp' checked >Reference SNP<br>
	     <input type='radio' name='region_method3' onclick='response_to_select_region(this)' value='gene'>Reference Gene<br>
	     <input type='radio' name='region_method3' onclick='response_to_select_region(this)' value='chr'>Chromosomal Region<br>
	    </td>
			</tr>
			<tr>
			<td class='region_detail_area'><table class='snp_region_spec'><tr>
					<td>Index SNP</td>
					<td><input type='text' name='refsnp3' size=15 /></td>
					</tr>
			
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='snpflank3' size=15 value=''/></td>
					</tr></table> <table class='gene_region_spec'>
			                <tr>
					<td>Reference Gene</td>
					<td><input type='text' name='refgene3' size=15 /></td>
					</tr>
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='geneflank3' size=15 value='' /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_gene3' size=15 /></td>
					</tr></table> <table class='chr_region_spec'>
					<tr>
					<td>Chromosome</td>
					<td><select name='chr3' ><option value='1'>1</option>
<option value='2'>2</option>
<option value='3'>3</option>
<option value='4'>4</option>
<option value='5'>5</option>
<option value='6'>6</option>
<option value='7'>7</option>
<option value='8'>8</option>
<option value='9'>9</option>
<option value='10'>10</option>
<option value='11'>11</option>
<option value='12'>12</option>
<option value='13'>13</option>
<option value='14'>14</option>
<option value='15'>15</option>
<option value='16'>16</option>
<option value='17'>17</option>
<option value='18'>18</option>
<option value='19'>19</option>
<option value='20'>20</option>
<option value='21'>21</option>
<option value='22'>22</option>
<option value='X'>X</option></select></td>
					</tr>
					<tr>
					<td>Start (Mb)</td>
					<td><input type='text' name='start3' size=15 /></td>
					<tr>
					<td>End (Mb)</td>
					<td><input type='text' name='end3' size=15 /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_chr3' size=15 /></td>
					</tr>
			</table> </td>
			</tr></table></td><td><table border=1 class='single_region_spec_head left_aln'>
			<tr>
				<td class='region_method_area'>
	    <input type='radio' name='region_method4' onclick='response_to_select_region(this)' value='snp' checked >Reference SNP<br>
	     <input type='radio' name='region_method4' onclick='response_to_select_region(this)' value='gene'>Reference Gene<br>
	     <input type='radio' name='region_method4' onclick='response_to_select_region(this)' value='chr'>Chromosomal Region<br>
	    </td>
			</tr>
			<tr>
			<td class='region_detail_area'><table class='snp_region_spec'><tr>
					<td>Index SNP</td>
					<td><input type='text' name='refsnp4' size=15 /></td>
					</tr>
			
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='snpflank4' size=15 value=''/></td>
					</tr></table> <table class='gene_region_spec'>
			                <tr>
					<td>Reference Gene</td>
					<td><input type='text' name='refgene4' size=15 /></td>
					</tr>
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='geneflank4' size=15 value='' /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_gene4' size=15 /></td>
					</tr></table> <table class='chr_region_spec'>
					<tr>
					<td>Chromosome</td>
					<td><select name='chr4' ><option value='1'>1</option>
<option value='2'>2</option>
<option value='3'>3</option>
<option value='4'>4</option>
<option value='5'>5</option>
<option value='6'>6</option>
<option value='7'>7</option>
<option value='8'>8</option>
<option value='9'>9</option>
<option value='10'>10</option>
<option value='11'>11</option>
<option value='12'>12</option>
<option value='13'>13</option>
<option value='14'>14</option>
<option value='15'>15</option>
<option value='16'>16</option>
<option value='17'>17</option>
<option value='18'>18</option>
<option value='19'>19</option>
<option value='20'>20</option>
<option value='21'>21</option>
<option value='22'>22</option>
<option value='X'>X</option></select></td>
					</tr>
					<tr>
					<td>Start (Mb)</td>
					<td><input type='text' name='start4' size=15 /></td>
					<tr>
					<td>End (Mb)</td>
					<td><input type='text' name='end4' size=15 /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_chr4' size=15 /></td>
					</tr>
			</table> </td>
			</tr></table></td><td><table border=1 class='single_region_spec_head left_aln'>
			<tr>
				<td class='region_method_area'>
	    <input type='radio' name='region_method5' onclick='response_to_select_region(this)' value='snp' checked >Reference SNP<br>
	     <input type='radio' name='region_method5' onclick='response_to_select_region(this)' value='gene'>Reference Gene<br>
	     <input type='radio' name='region_method5' onclick='response_to_select_region(this)' value='chr'>Chromosomal Region<br>
	    </td>
			</tr>
			<tr>
			<td class='region_detail_area'><table class='snp_region_spec'><tr>
					<td>Index SNP</td>
					<td><input type='text' name='refsnp5' size=15 /></td>
					</tr>
			
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='snpflank5' size=15 value=''/></td>
					</tr></table> <table class='gene_region_spec'>
			                <tr>
					<td>Reference Gene</td>
					<td><input type='text' name='refgene5' size=15 /></td>
					</tr>
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='geneflank5' size=15 value='' /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_gene5' size=15 /></td>
					</tr></table> <table class='chr_region_spec'>
					<tr>
					<td>Chromosome</td>
					<td><select name='chr5' ><option value='1'>1</option>
<option value='2'>2</option>
<option value='3'>3</option>
<option value='4'>4</option>
<option value='5'>5</option>
<option value='6'>6</option>
<option value='7'>7</option>
<option value='8'>8</option>
<option value='9'>9</option>
<option value='10'>10</option>
<option value='11'>11</option>
<option value='12'>12</option>
<option value='13'>13</option>
<option value='14'>14</option>
<option value='15'>15</option>
<option value='16'>16</option>
<option value='17'>17</option>
<option value='18'>18</option>
<option value='19'>19</option>
<option value='20'>20</option>
<option value='21'>21</option>
<option value='22'>22</option>
<option value='X'>X</option></select></td>
					</tr>
					<tr>
					<td>Start (Mb)</td>
					<td><input type='text' name='start5' size=15 /></td>
					<tr>
					<td>End (Mb)</td>
					<td><input type='text' name='end5' size=15 /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_chr5' size=15 /></td>
					</tr>
			</table> </td>
			</tr></table></td></tr><tr><td><table border=1 class='single_region_spec_head left_aln'>
			<tr>
				<td class='region_method_area'>
	    <input type='radio' name='region_method6' onclick='response_to_select_region(this)' value='snp' checked >Reference SNP<br>
	     <input type='radio' name='region_method6' onclick='response_to_select_region(this)' value='gene'>Reference Gene<br>
	     <input type='radio' name='region_method6' onclick='response_to_select_region(this)' value='chr'>Chromosomal Region<br>
	    </td>
			</tr>
			<tr>
			<td class='region_detail_area'><table class='snp_region_spec'><tr>
					<td>Index SNP</td>
					<td><input type='text' name='refsnp6' size=15 /></td>
					</tr>
			
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='snpflank6' size=15 value=''/></td>
					</tr></table> <table class='gene_region_spec'>
			                <tr>
					<td>Reference Gene</td>
					<td><input type='text' name='refgene6' size=15 /></td>
					</tr>
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='geneflank6' size=15 value='' /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_gene6' size=15 /></td>
					</tr></table> <table class='chr_region_spec'>
					<tr>
					<td>Chromosome</td>
					<td><select name='chr6' ><option value='1'>1</option>
<option value='2'>2</option>
<option value='3'>3</option>
<option value='4'>4</option>
<option value='5'>5</option>
<option value='6'>6</option>
<option value='7'>7</option>
<option value='8'>8</option>
<option value='9'>9</option>
<option value='10'>10</option>
<option value='11'>11</option>
<option value='12'>12</option>
<option value='13'>13</option>
<option value='14'>14</option>
<option value='15'>15</option>
<option value='16'>16</option>
<option value='17'>17</option>
<option value='18'>18</option>
<option value='19'>19</option>
<option value='20'>20</option>
<option value='21'>21</option>
<option value='22'>22</option>
<option value='X'>X</option></select></td>
					</tr>
					<tr>
					<td>Start (Mb)</td>
					<td><input type='text' name='start6' size=15 /></td>
					<tr>
					<td>End (Mb)</td>
					<td><input type='text' name='end6' size=15 /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_chr6' size=15 /></td>
					</tr>
			</table> </td>
			</tr></table></td><td><table border=1 class='single_region_spec_head left_aln'>
			<tr>
				<td class='region_method_area'>
	    <input type='radio' name='region_method7' onclick='response_to_select_region(this)' value='snp' checked >Reference SNP<br>
	     <input type='radio' name='region_method7' onclick='response_to_select_region(this)' value='gene'>Reference Gene<br>
	     <input type='radio' name='region_method7' onclick='response_to_select_region(this)' value='chr'>Chromosomal Region<br>
	    </td>
			</tr>
			<tr>
			<td class='region_detail_area'><table class='snp_region_spec'><tr>
					<td>Index SNP</td>
					<td><input type='text' name='refsnp7' size=15 /></td>
					</tr>
			
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='snpflank7' size=15 value=''/></td>
					</tr></table> <table class='gene_region_spec'>
			                <tr>
					<td>Reference Gene</td>
					<td><input type='text' name='refgene7' size=15 /></td>
					</tr>
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='geneflank7' size=15 value='' /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_gene7' size=15 /></td>
					</tr></table> <table class='chr_region_spec'>
					<tr>
					<td>Chromosome</td>
					<td><select name='chr7' ><option value='1'>1</option>
<option value='2'>2</option>
<option value='3'>3</option>
<option value='4'>4</option>
<option value='5'>5</option>
<option value='6'>6</option>
<option value='7'>7</option>
<option value='8'>8</option>
<option value='9'>9</option>
<option value='10'>10</option>
<option value='11'>11</option>
<option value='12'>12</option>
<option value='13'>13</option>
<option value='14'>14</option>
<option value='15'>15</option>
<option value='16'>16</option>
<option value='17'>17</option>
<option value='18'>18</option>
<option value='19'>19</option>
<option value='20'>20</option>
<option value='21'>21</option>
<option value='22'>22</option>
<option value='X'>X</option></select></td>
					</tr>
					<tr>
					<td>Start (Mb)</td>
					<td><input type='text' name='start7' size=15 /></td>
					<tr>
					<td>End (Mb)</td>
					<td><input type='text' name='end7' size=15 /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_chr7' size=15 /></td>
					</tr>
			</table> </td>
			</tr></table></td><td><table border=1 class='single_region_spec_head left_aln'>
			<tr>
				<td class='region_method_area'>
	    <input type='radio' name='region_method8' onclick='response_to_select_region(this)' value='snp' checked >Reference SNP<br>
	     <input type='radio' name='region_method8' onclick='response_to_select_region(this)' value='gene'>Reference Gene<br>
	     <input type='radio' name='region_method8' onclick='response_to_select_region(this)' value='chr'>Chromosomal Region<br>
	    </td>
			</tr>
			<tr>
			<td class='region_detail_area'><table class='snp_region_spec'><tr>
					<td>Index SNP</td>
					<td><input type='text' name='refsnp8' size=15 /></td>
					</tr>
			
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='snpflank8' size=15 value=''/></td>
					</tr></table> <table class='gene_region_spec'>
			                <tr>
					<td>Reference Gene</td>
					<td><input type='text' name='refgene8' size=15 /></td>
					</tr>
					<tr>
					<td>Flanking region (kb)</td>
					<td><input type='text' name='geneflank8' size=15 value='' /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_gene8' size=15 /></td>
					</tr></table> <table class='chr_region_spec'>
					<tr>
					<td>Chromosome</td>
					<td><select name='chr8' ><option value='1'>1</option>
<option value='2'>2</option>
<option value='3'>3</option>
<option value='4'>4</option>
<option value='5'>5</option>
<option value='6'>6</option>
<option value='7'>7</option>
<option value='8'>8</option>
<option value='9'>9</option>
<option value='10'>10</option>
<option value='11'>11</option>
<option value='12'>12</option>
<option value='13'>13</option>
<option value='14'>14</option>
<option value='15'>15</option>
<option value='16'>16</option>
<option value='17'>17</option>
<option value='18'>18</option>
<option value='19'>19</option>
<option value='20'>20</option>
<option value='21'>21</option>
<option value='22'>22</option>
<option value='X'>X</option></select></td>
					</tr>
					<tr>
					<td>Start (Mb)</td>
					<td><input type='text' name='start8' size=15 /></td>
					<tr>
					<td>End (Mb)</td>
					<td><input type='text' name='end8' size=15 /></td>
					<tr>
					<td>Optional Index SNP (default is SNP with lowest P value)</td>
					<td><input type='text' name='refsnp_for_chr8' size=15 /></td>
					</tr>
			</table> </td>
			</tr></table></td></tr></table></div></td>
			</tr></table></div></td>
		</tr>
	</table>
</div>
<hr>
<h2>
	<span title='Plot HiC interaction heatmap'>HiC interaction plot</span>
</h2>
<div class="form-group">
<div class="checkbox"><span title='Do you want an HiC interaction heatmap plot?'><label>
	 <input type="checkbox" name="heatmap_toggle" value="on" checked="checked" id="heatmap_toggle_id" />Output interaction heatmap?
</label>
</span>
</div>
		<td>Interaction type (resolution)</td><td><label>
		<input type="radio" name="interaction_type" id="interaction_type_inter_id" value="interchromosomal" onclick="$('#interaction_chr_tr_id').show();" >INTERchromosomal(1Mb)
		</label><br>
		<input type="radio" id="interaction_type_intra_id" name="interaction_type" value="intrachromosomal" onclick="$('#interaction_chr_tr_id').hide();" checked="checked">INTRAchromosomal(100Kb)
		</label></td>
	</tr>
	<tr>
		<td>Cell lines</td><td><label>
		<input type="radio" name="interaction_cell_type" id="interaction_cell_type_k562_id" value="k562" >K562
		</label>
		<input type="radio" name="interaction_cell_type" id="interaction_cell_type_gm06690_id" value="gm06690" checked="checked">GM06690
		</label></td>
	</tr>
	<tr id="interaction_chr_tr_id" style="display:none">
		<td>Chromosome</td><td><select name='interaction_chr' ><option value='1'>1</option>
<option value='2'>2</option>
<option value='3'>3</option>
<option value='4'>4</option>
<option value='5'>5</option>
<option value='6'>6</option>
<option value='7'>7</option>
<option value='8'>8</option>
<option value='9'>9</option>
<option value='10'>10</option>
<option value='11'>11</option>
<option value='12'>12</option>
<option value='13'>13</option>
<option value='14'>14</option>
<option value='15'>15</option>
<option value='16'>16</option>
<option value='17'>17</option>
<option value='18'>18</option>
<option value='19'>19</option>
<option value='20'>20</option>
<option value='21'>21</option>
<option value='22'>22</option>
<option value='X'>X</option></select></td>
</div>
<hr>
<br> <br>
<h2>
	<span title='show signal strengths in regions'>Generic plot (using UCSC BED tables)</span>
</h2>
<table class="noborder">
	<tr>
		<td colspan="2"><span title='Do you want a generic (annotation) plot?'><label>
	 <input type="checkbox" name="generic_toggle" value="on" checked="checked" id="generic_toggle_id" />Generic plot?
</label>
</span></td>
	</tr>
	<tr>
		<td colspan="2"><span title='Do you want text annotation?'><label>
	 <input type="checkbox" name="anno_toggle" value="on" id="anno_toggle_id" />Output ANNOVAR annotation?
</label>
</span></td>
	</tr>
	<tr>
		<td colspan="2"><span title='First 5 columns correspond to chromosome,start,end,alternative allele,reference allele'><label>
	 <input type="checkbox" name="avinput" value="on" id="avinput_id" />Input file in ANNOVAR format?
</label>
</span></td>
	</tr>
	<tr>
		<td>Advanced options</td><td><label>
        	<input type="radio" name="detail_toggle" value="show" onclick="showDetail()">show
        	</label>
        	<input type="radio" name="detail_toggle" value="hide" onclick="hideDetail()" checked="checked">hide
        	</label></td>
	</tr>
</table>
<table class="advanced" style="display:none" id="option_detail_id">
	<tr>
		<td><label>
	 <input type="checkbox" name="ld_toggle" value="on" checked="checked" id="ld_toggle_id" />Output linkage disequilibrium (only works with SINGLE region, written in input file)
</label></td>
	</tr>
</table>
</br></br><p class="center">
	<b>Please upload your own files <b style="color:red">AFTER</b> selecting data tracks.</b><br>
</p>
<table>
	<tr>
		<th>
			Cell Line
		</th>
		<th>
			Experiment Type
		</th>
		<th>
			Data Tracks (max: 20; <a class='button' href="/example/example.bed"><strong>BED Example</strong></a>)
		</th>
	</tr>
	<tr>
		<td class="table_align"><table class="noborder left_aln">
			<tr>
				<td><span title='Human lung carcinoma cell line
'><label>
	 <input type="checkbox" name="" value="A549" id="A549" onchange="changeTracks()" class="cell" />A549
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='Human colorectal adenocarcinoma cell line
'><label>
	 <input type="checkbox" name="" value="Caco-2" id="Caco-2" onchange="changeTracks()" class="cell" />Caco-2
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='Human lymphoblastoid cell line
'><label>
	 <input type="checkbox" name="" value="GM12878" class="cell" id="GM12878" onchange="changeTracks()" />GM12878
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='H1 human embryonic stem cells
'><label>
	 <input type="checkbox" name="" value="H1-hESC" class="cell" onchange="changeTracks()" id="H1-hESC" />H1-hESC
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='Human colorectal carcinoma
'><label>
	 <input type="checkbox" name="" value="HCT116" id="HCT116" onchange="changeTracks()" class="cell" />HCT116
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='Human Mammary Epithelial Cells
'><label>
	 <input type="checkbox" name="" value="HMEC" class="cell" id="HMEC" onchange="changeTracks()" />HMEC
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='Human skeletal muscle myoblasts
'><label>
	 <input type="checkbox" name="" value="HSMM" onchange="changeTracks()" id="HSMM" class="cell" />HSMM
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='Human umbilical vein endothelial cell
'><label>
	 <input type="checkbox" name="" value="HUVEC" id="HUVEC" onchange="changeTracks()" class="cell" />HUVEC
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='Human cervical cancer cell line
'><label>
	 <input type="checkbox" name="" value="HeLa-S3" class="cell" id="HeLa-S3" onchange="changeTracks()" />HeLa-S3
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='Human hepatoma cell line
'><label>
	 <input type="checkbox" name="" value="HepG2" class="cell" id="HepG2" onchange="changeTracks()" />HepG2
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='Human chronic myelogenous leukemia cell line
'><label>
	 <input type="checkbox" name="" value="K562" id="K562" onchange="changeTracks()" class="cell" />K562
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='Human breast cancer cell line
'><label>
	 <input type="checkbox" name="" value="MCF-7" class="cell" id="MCF-7" onchange="changeTracks()" />MCF-7
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='Normal Human Epidermal Keratinocytes
'><label>
	 <input type="checkbox" name="" value="NHEK" onchange="changeTracks()" id="NHEK" class="cell" />NHEK
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='Normal Human Lung Fibroblasts
'><label>
	 <input type="checkbox" name="" value="NHLF" class="cell" id="NHLF" onchange="changeTracks()" />NHLF
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='Not a specific cell line
'><label>
	 <input type="checkbox" name="" value="None" class="cell" onchange="changeTracks()" id="None" />None
</label>
</span></td>
			</tr>
		</table></td> <td class="table_align"><table class="noborder left_aln">
			<tr>
				<td><span title='This track reports the enrichment of CTCF binding measured by ChIP-seq. CTCF is a transcription regulator that could bind to insulator sequence.
'><label>
	 <input type="checkbox" name="" value="ChIP-seq_CTCF" id="ChIP-seq_CTCF" onchange="changeTracks()" class="experiment" />ChIP-seq_CTCF
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track reports the enrichment of an epigenetic marker measured by ChIP-seq.
'><label>
	 <input type="checkbox" name="" value="ChIP-seq_H3K27ac" class="experiment" id="ChIP-seq_H3K27ac" onchange="changeTracks()" />ChIP-seq_H3K27ac
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track reports the enrichment of an epigenetic marker measured by ChIP-seq.
'><label>
	 <input type="checkbox" name="" value="ChIP-seq_H3K27me3" onchange="changeTracks()" id="ChIP-seq_H3K27me3" class="experiment" />ChIP-seq_H3K27me3
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track reports the enrichment of an epigenetic marker measured by ChIP-seq.
'><label>
	 <input type="checkbox" name="" value="ChIP-seq_H3K36me3" class="experiment" id="ChIP-seq_H3K36me3" onchange="changeTracks()" />ChIP-seq_H3K36me3
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track reports the enrichment of an epigenetic marker measured by ChIP-seq.
'><label>
	 <input type="checkbox" name="" value="ChIP-seq_H3K4me1" onchange="changeTracks()" id="ChIP-seq_H3K4me1" class="experiment" />ChIP-seq_H3K4me1
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track reports the enrichment of an epigenetic marker measured by ChIP-seq.
'><label>
	 <input type="checkbox" name="" value="ChIP-seq_H3K4me3" class="experiment" onchange="changeTracks()" id="ChIP-seq_H3K4me3" />ChIP-seq_H3K4me3
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track reports the enrichment of an epigenetic marker measured by ChIP-seq.
'><label>
	 <input type="checkbox" name="" value="ChIP-seq_H3K79me2" class="experiment" id="ChIP-seq_H3K79me2" onchange="changeTracks()" />ChIP-seq_H3K79me2
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track reports the enrichment of an epigenetic marker measured by ChIP-seq.
'><label>
	 <input type="checkbox" name="" value="ChIP-seq_H3K9ac" onchange="changeTracks()" id="ChIP-seq_H3K9ac" class="experiment" />ChIP-seq_H3K9ac
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track reports the enrichment of an epigenetic marker measured by ChIP-seq.
'><label>
	 <input type="checkbox" name="" value="ChIP-seq_H3K9me3" class="experiment" id="ChIP-seq_H3K9me3" onchange="changeTracks()" />ChIP-seq_H3K9me3
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track reports the enrichment of TCF3 binding measured by ChIP-seq. TCF3 is a transcription factor.
'><label>
	 <input type="checkbox" name="" value="ChIP-seq_TCF3" class="experiment" id="ChIP-seq_TCF3" onchange="changeTracks()" />ChIP-seq_TCF3
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track reports the enrichment of TCF7L2 binding measured by ChIP-seq. TCF7L2 is a transcription factor in Wnt pathway.
'><label>
	 <input type="checkbox" name="" value="ChIP-seq_TCF7L2" class="experiment" id="ChIP-seq_TCF7L2" onchange="changeTracks()" />ChIP-seq_TCF7L2
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track reports DNase Hyper-sensitivity Sites measured by DNase-seq.
'><label>
	 <input type="checkbox" name="" value="DNase-seq" onchange="changeTracks()" id="DNase-seq" class="experiment" />DNase-seq
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track reports nucleosome-depleted regions of the genome measured by FAIRE-seq.
'><label>
	 <input type="checkbox" name="" value="FAIRE-seq" id="FAIRE-seq" onchange="changeTracks()" class="experiment" />FAIRE-seq
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track contains the location and score of transcription factor binding sites conserved in the human/mouse/rat alignment.
'><label>
	 <input type="checkbox" name="" value="TFBS Conservation" id="TFBS Conservation" onchange="changeTracks()" class="experiment" />TFBS Conservation
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='"This track shows regions where transcription factors, proteins responsible for modulating gene transcription, bind to DNA as assayed by ChIP-seq."
'><label>
	 <input type="checkbox" name="" value="TFBS Region" id="TFBS Region" onchange="changeTracks()" class="experiment" />TFBS Region
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='This track displays a chromatin state segmentation for each cell line. The states were computed integrating multiple ChIP-seq data sets using HMM method.
'><label>
	 <input type="checkbox" name="" value="chromHMM" class="experiment" id="chromHMM" onchange="changeTracks()" />chromHMM
</label>
</span></td>
			</tr>
			<tr>
				<td><span title='"The track reports the percentage of DNA molecules that exhibit cytosine methylation at specific CpG dinucleotides. The score in this track reports the number of sequencing reads obtained for each CpG, which is often called 'coverage'. The score is capped at 1000, so any CpGs that were covered by more than 1000 sequencing reads have a score of 1000."
'><label>
	 <input type="checkbox" name="" value="methyl RRBS" onchange="changeTracks()" id="methyl RRBS" class="experiment" />methyl RRBS
</label>
</span></td>
			</tr>
		</table></td> <td class="table_align"><table class="noborder left_aln" id="dataTrackHere">
	<p>
		
	</p>
</table></td>
	</tr>
</table>
<table class="noborder">
	<tr>
		<td><p>
	<input type="submit" class="btn btn-primary btn-lg" name="Submit" value="Submit" /> <input class="btn btn-primary btn-lg" type="reset"  name=".reset" />
</p></td>
	</tr>
</table>
<div><input type="hidden" name=".cgifields" value="generic_toggle"  /><input type="hidden" name=".cgifields" value="qformat"  /><input type="hidden" name=".cgifields" value="anno_toggle"  /><input type="hidden" name=".cgifields" value="ld_toggle"  /><input type="hidden" name=".cgifields" value=""  /><input type="hidden" name=".cgifields" value="avinput"  /><input type="hidden" name=".cgifields" value="heatmap_toggle"  /></div>
</form><p>
	Please send questions or comments to <strong>yunfeigu@usc.edu</strong>
</p>
      <hr>

      <footer>
        <p>&copy; Wang Genomics Lab 2013-2014</p>
      </footer>
    </div> <!-- /container -->


    <!-- Bootstrap core JavaScript
    ================================================== -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
    <script src="js/bootstrap.min.js"></script>
  </body>
</html>
