<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>Enlight</title>
<meta name="keywords" content="green jelly, theme, free templates, nivo image slider, website, templatemo, CSS, HTML" />
<meta name="description" content="Green Jelly Theme, free CSS template provided by templatemo.com" />
<link href="../css/templatemo_style.css" rel="stylesheet" type="text/css" />

<link rel="stylesheet" href="../css/nivo-slider.css" type="text/css" media="screen" />

<script language="javascript" type="text/javascript">
function clearText(field)
{
    if (field.defaultValue == field.value) field.value = '';
    else if (field.value == '') field.value = field.defaultValue;
}
</script>

<link rel="stylesheet" type="text/css" href="../css/ddsmoothmenu.css" />

<script type="text/javascript" src="../js/jquery.min.js"></script>
<script type="text/javascript" src="../js/ddsmoothmenu.js">

/***********************************************
* Smooth Navigational Menu- (c) Dynamic Drive DHTML code library (www.dynamicdrive.com)
* This notice MUST stay intact for legal use
* Visit Dynamic Drive at http://www.dynamicdrive.com/ for full source code
***********************************************/

</script>

<script type="text/javascript">

ddsmoothmenu.init({
	mainmenuid: "templatemo_menu", //menu DIV id
	orientation: 'h', //Horizontal or vertical menu: Set to "h" or "v"
	classname: 'ddsmoothmenu', //class added to menu's outer DIV
	//customtheme: ["#1c5a80", "#18374a"],
	contentsource: "markup" //"markup" or ["container_id", "path_to_menu_file"]
})

</script>
  
</head>
<body>

<div id="templatemo_wrapper">
	<div id="templatemo_header"></div> <!-- end of header -->
      <div id="templatemo_menu" class="ddsmoothmenu">
        <ul>
            <li><a href="../index.html" >Home</a></li>
            <li><a href="pipeline.html">Pipeline</a> </li>
          	<li><a href="help.html" class="selected">Help</a></li>
            </ul>
        <br style="clear: left" />
    </div> <!-- end of menu -->
    
    <script type="text/javascript" src="../js/jquery-1.4.3.min.js"></script>
    <script type="text/javascript" src="../js/jquery.nivo.slider.js"></script>
    <script type="text/javascript">
    $(window).load(function() {
    $('#slider').nivoSlider();
    });
    </script>
    
    <div id="templatemo_main"><br class="cleaner" />

<p><b>Genotype</b></p>
<p>The genotype is the genetic makeup of a cell, an organism, or an individual (i.e. the specific allele makeup of the individual) usually with reference to a specific character under consideration. More information please refers to <a href="http://www.answers.com/topic/genotype" target="_blank"><i><u>http://www.answers.com/topic/genotype</u></i></a>.</p>
<p><b>Consensus sequence</b></p>
<p>In molecular biology and bioinformatics, consensus sequence refers to the most common nucleotide or amino acid at a particular position after multiple sequences are aligned. A consensus sequence is a way of representing the results of a multiple sequence alignment, where related sequences are compared to each other, and similar functional sequence motifs are found. The conserved sequence motifs are called consensus sequences and they show which residues are conserved and which residues are variable. The consensus sequence also shows which residues are most abundant in the alignment at each position. More information please refers to <a href="http://en.mimi.hu/biology/consensus_sequence.html" target="_blank"><i><u>http://en.mimi.hu/biology/consensus_sequence.html</u></i></a>.</p>
<p><b>Intergenic region</b></p>
<p>An Intergenic region (IGR) is a stretch of DNA sequences located between clusters of genes that contain few or no genes. Occasionally some intergenic DNA acts to control genes nearby, but most of it has no currently known function. It is one of the DNA sequences collectively referred to as junk DNA, though it is only one phenomenon labeled such and in scientific studies today, the term is less used. In humans, intergenic regions comprise a large percentage of the genome. This could also be where non-coding RNAs are located. Though little is known about them, they are thought to have regulatory functions. More information please refers to <a href="http://www.answers.com/topic/intergenic-region" target="_blank"><i><u>http://www.answers.com/topic/intergenic-region</u></i></a>.</p>
<p><b>UTR</b></p>
<p>In molecular genetics, an untranslated region (or UTR) refers to either of two sections on each side of a coding sequenceon a strand of mRNA. If it is found on the 5' side, it is called the 5' UTR (or leader sequence), or if it is found on the 3' side, it is called the 3' UTR (or trailer sequence). More information please refers to <a href="http://www.answers.com/topic/untranslated-region" target="_blank"><i><u>http://www.answers.com/topic/untranslated-region</u></i></a>.</p>
<p><b>Locus</b></p>
<p>In the fields of genetics and evolutionary computation, a locus (plural loci) is the specific location of a gene or DNA sequence on a chromosome. A variant of the DNA sequence at a given locus is called an allele. The ordered list of loci known for a particular genome is called a genetic map. Gene mapping is the process of determining the locus for a particular biological trait. Diploid and polyploidy cells whose chromosomes have the same allele of a given gene at some locus are called homozygous with respect to that gene, while those that have different alleles of a given gene at a locus, heterozygous with respect to that gene. More information please refers to <a href="http://en.mimi.hu/biology/locus.html" target="_blank"><i><u>http://en.mimi.hu/biology/locus.html</u></i></a>.</p>
<p><b>SNP</b></p>
<p>A single-nucleotide polymorphism (SNP, pronounced snip) is a DNA variation occurring when a single nucleotide- A, T, C, or G- in the genome (or other shared sequence) differs between members of a biological species or paired chromosomes in an individual. More information please refers to <a href="http://www.ncbi.nlm.nih.gov/About/primer/snps.html" target="_blank"><i><u>http://www.ncbi.nlm.nih.gov/About/primer/snps.html</u></i></a>.</p>
<p><b>InDel</b></p>
<p>Indel is a molecular biology term that has different definitions in different fields: in evolutionary studies, indel is used to mean an insertion or a deletion and indels simply refers to the mutation class that includes both insertions, deletions, and the combination thereof, including insertion and deletion events that may be separated by many years. In germ line and somatic mutation studies, however, indel describes a special mutation class, defined as a mutation resulting in a colocalized insertion and deletion and a net gain or loss in nucleotides, and microindel is defined as an indel that results in a net gain or loss of 1 to 50 nucleotides. More information please refers to <a href="http://www.answers.com/topic/indel" target="_blank"><i><u>http://www.answers.com/topic/indel</u></i></a>.</p>


    </div> <!-- end of main -->
    
    <div id="templatemo_footer">
      <div class="cleaner"></div>
    </div> <!-- end of footer -->
</div>

<div id="templatemo_cr_bar_wrapper">
	<div id="templatemo_cr_bar">
    	Copyright Â© 2013 <a href="http://genomics.usc.edu">USC Wang Lab</a> | Designed by <a href="http://www.templatemo.com" target="_parent">Free CSS Templates</a>
    </div>
</div>


<script type='text/javascript' src='../js/logging.js'></script>
</body>
</html>
