<!DOCTYPE html>
<html lang="en">
    <head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<meta name="description" content="">
	<meta name="author" content="">
	<link rel="icon" href="../images/favicon.ico">

	<title>Enlight</title>

	<!-- Bootstrap core CSS -->
	<link href="../css/bootstrap.min.css" rel="stylesheet">

	<!-- Custom styles for this template -->
	<link href="../css/jumbotron.css" rel="stylesheet">
	<link href="../css/custom.css" rel="stylesheet">

	<!-- IE10 viewport hack for Surface/desktop Windows 8 bug -->
	<script src="../js/ie10-viewport-bug-workaround.js"></script>

	<!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
	<!--[if lt IE 9]>
	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
	<![endif]-->

	<script>
	    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
		    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
		m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
	    
	    ga('create', 'UA-50646209-1', 'usc.edu');
	    ga('send', 'pageview');
	    
	</script>

    </head>

    <body>

	<div class="navbar navbar_custom navbar-fixed-top" role="navigation">
	    <div class="container">
		<div class="navbar-header">
		    <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
			<span class="sr-only">Toggle navigation</span>
			<span class="icon-bar"></span>
			<span class="icon-bar"></span>
			<span class="icon-bar"></span>
		    </button>
	  <a class="navbar-brand" style="padding:5px" href="/index.html"> <img src="/images/enlight.png" style="height:100%"></a>
        </div>
	<div class="navbar-collapse collapse">
	    <ul class="nav navbar-nav">
		<li><a class="navbar_custom" href="../index.html">Home</a></li>
		<li><a class="navbar_custom" href="tutorial.html">Tutorial</a></li>
		<li><a class="active navbar_custom" href="help.html">Help</a></li>
		<li><a class="navbar_custom" href="faq.html">FAQ</a></li>
		<li><a class="navbar_custom" href="example.html">Example</a></li>
		<li class="dropdown">
		<a href="#" class="navbar_custom dropdown-toggle" data-toggle="dropdown">Related projects <span class="caret"></span></a>
		<ul class="dropdown-menu" role="menu">
		    <li><a href="http://icages.usc.edu:5000">iCAGES</a></li>
		    <li><a href="http://phenolyzer.usc.edu">Phenolyzer</a></li>
		    <li><a href="http://wannovar.usc.edu">wANNOVAR</a></li>
		</ul>
		</li>
		    </ul>
		    <div class="navbar-header navbar-right col-md-3" >
			<a class="title navbar-brand" href="http://genomics.usc.edu" style="padding:5px"><img src="../images/wanglablogo.png" alt="" style="height:100%;"></a>
		    </div>
		</div><!--/.nav-collapse -->
	    </div>
	</div>

	<div class="container">
	    <h2>Server Architecture</h2>
	    <p>
Enlight takes at least two columns of input, the SNP name and P value. After submission, a job will be scheduled in MySQL database. Unpon execution, a modified version of LocusZoom will output regional plots and annotation plots; meanwhile, ANNOVAR outputs text annotation for each variant.
	    </p>
	    <img class='center-block' src="../images/server_architecture.png" width=670 height=460>
	<h2 id="exampleinput">Example Input</h2>
	<img width=600 height=340 src="../images/enlight_example_input.png" class='center'>
	<h2 id="exampleoutput_plot">Example Output (plot)</h2>
	<p>
	A typical plot consists of 3 parts: regional plot plus annotation plots, HiC interaction heatmap and summary plots. In this example, the first part shows the association signals for rs2071278 and nearby loci from Well-come Trust Case Control Consortium (WTCCC) Rheumatoid Arthritis study. The color stripe shows the chromHMM segmentation result which combines multiple epigenetic chromatin marks (legend shown below). The three SNPs rs204991, rs204990, rs204989 (all yellow), which are in high LD with the reference SNP, show up in a proposed enhancer region in blood cell lines, GM12878 and K562, but not in non-blood cell lines, HepG2 or HSMM. The UChicago_eQTL panel shows association signal based on data from eQTL browser at University of Chicago. The wgEncodeRegTfbsClusteredV3 panel summarizes a large collection of transcription factor ChIP-seq results, and tfbsConsSites panel describes TFBS conservation across human/mouse/rat.</p>
	<p>rs2071278 is genome-wide significantly associated with serum complement C3 and C4 levels (Yang, et al., 2012), important measurements in assessment of rheumatoid arthritis (Makinde, et al., 1989). Therefore, rs2071278 is possibly a proxy SNP tagging nearby functional variants. For instance, three SNPs rs204991, rs204990, rs204989 (yellow), all in high LD with rs2071278 (1000 genomes, European population, 2012), are shown to be significantly associated with expression of human leukocyte antigen (HLA) genes by microarray (Zeller, et al., 2010). This finding partly explains their strong associations with rheumatoid arthritis in WTCCC study(Wellcome Trust Case Control, 2007). When strong associations appear near known GWAS hit and are surrounded by interesting epigenetic and other functional features, they perhaps suggest functional importance.
	<p>
	The explanations for the 2nd and 3rd parts are shown alongside the figures.
	</p>
	<br />
	<h3 id="enlight_example_output">Regional plot plus annotation plots</h3>
	<img width=1150 height=1200 src="../images/enlight_example_output.png" class='center'>
	<h3 id="enlight_example_output_heatmap">HiC interaction heatmap</h3>
	<img width=1150 height=1200 src="../images/enlight_example_output_heatmap.png" class='center'>
	<h3 id="enlight_example_output_summary">Summary plots</h3>
	<img width=1150 height=1200 src="../images/enlight_example_output_summary.png" class='center'>
	<h2 id="exampleoutput_text">Example Text Annotation</h2>
	<p><b>If you opt to output ANNOVAR annotation, you will get a '*.hg19_multianno.html' on result page. This is the text annotation for each variant in the input file. ANNOVAR will output gene annotation, minor allele frequency (MAF) from 1000 Genomes project as well as selected UCSC datatracks and other annotation info (eg custome BED file, eQTL etc). If your input file does not have the 5 columns required by ANNOVAR (chr,start,end,alt,ref), Enlight will first convert rsIDs to chromosomal positions according to dbSNP137, and then run annotation.  Sometimes, you will see *.invalid_input files on result page, they contain lines that can't by recognized by ANNOVAR, e.g. lines that don't have chr/start/end.</b></p>
	<br />
	<img  width=900 height=300 src="../images/enlight_example_text_output.png" class='center'>
	<p>NOTE: The above output has been adjusted for display purposes. Actual results will differ.</p>
	<br /><br />
	</div><!--panel-group-->
	<hr>
	<footer>
	<p>&copy; Wang Genomics Lab 2013-2014</p>
	</footer>
    </div> <!-- /container -->


    <!-- Bootstrap core JavaScript
    ================================================== -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
    <script src="../js/bootstrap.min.js"></script>
</body>
</html>
