<!DOCTYPE html>
<html lang="en">
    <head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<meta name="description" content="">
	<meta name="author" content="">
	<link rel="icon" href="images/favicon.ico">

	<title>Enlight</title>

	<!-- Bootstrap core CSS -->
	<link href="../css/bootstrap.min.css" rel="stylesheet">

	<!-- Custom styles for this template -->
	<link href="../css/jumbotron.css" rel="stylesheet">
	<link href="../css/custom.css" rel="stylesheet">

	<!-- IE10 viewport hack for Surface/desktop Windows 8 bug -->
	<script src="../js/ie10-viewport-bug-workaround.js"></script>

	<!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
	<!--[if lt IE 9]>
	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
	<![endif]-->

	<script>
	    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
		    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
		m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
	    
	    ga('create', 'UA-50646209-1', 'usc.edu');
	    ga('send', 'pageview');
	    
	</script>

    </head>

    <body>

	<div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
	    <div class="container">
		<div class="navbar-header">
		    <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
			<span class="sr-only">Toggle navigation</span>
			<span class="icon-bar"></span>
			<span class="icon-bar"></span>
			<span class="icon-bar"></span>
		    </button>
	  <a class="navbar-brand navbar_custom" href="../index.html">Enlight</a>
        </div>
	<div class="navbar-collapse collapse">
	    <ul class="nav navbar-nav">
		<li class="navbar_custom"><a href="../index.html">Home</a></li>
		<li><a class="navbar_custom" href="tutorial.html">Tutorial</a></li>
		<li><a class="navbar_custom" href="help.html">Help</a></li>
		<li><a class="navbar_custom" href="pipeline.html">Pipeline</a></li>
		<li><a class="active navbar_custom" href="faq.html">FAQ</a></li>
		<li class="dropdown">
		<a href="#" class="navbar_custom dropdown-toggle" data-toggle="dropdown">Example <span class="caret"></span></a>
			<ul class="dropdown-menu" role="menu">
			    <li><a href="example/exampleinput.html">Example Input</a></li>
			    <li><a href="example/exampleoutput/index.html">Example Output</a></li>
			</ul>
			</li>
		    </ul>
	    <div class="navbar-header navbar-right col-md-3" >
		<a class="title navbar-brand" href="http://genomics.usc.edu"><img src="../images/wanglablogo.png" alt="" style="height:100%;"></a>
	    </div>
		</div><!--/.nav-collapse -->
	    </div>
	</div>

	<div class="container">
	    <h2>Enlight FAQ</h2>
	    <div class="panel-group">
		<div class="panel panel-default">
		    <div class="panel-heading">
			<h4 class="panel-title">
			    <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion" href="#q1">Where does the eQTL data come from? How is it compiled?</a>
			</h4>
		    </div>
		    <div id="q1" class="panel-collapse collapse">
			<div class="panel-body">
			    <p>The eQTL data sets come from two sources. The first one is the <a href="http://eqtl.uchicago.edu/cgi-bin/gbrowse/eqtl/">eQTL browser</a> of Pritchard Lab at University of Chicago, the other one is the <a href="http://www.ncbi.nlm.nih.gov/projects/gap/eqtl/index.cgi">GTEx project</a>. The GTEx eQTL data is split into different subsets based on analysis IDs. Analysis IDs usually represent different tissues. However, sometimes they merely represent different measurements for expression. The name of a data set can tell you the tissue and analysis assay (if more than one are assays are used). Tissue-specific information is unavailable for data from University of Chicago as no such meta-data is provided.</p>
			</div>
		    </div>
		</div>
		<div class="panel panel-default">
		    <div class="panel-heading">
			<h4 class="panel-title">
			    <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion" href="#q2">Where does the ENCODE data come from? How is it compiled?</a>
			</h4>
		    </div>
		    <div id="q2" class="panel-collapse collapse">
			<div class="panel-body">
			    <p>All ENCODE data sets come from UCSC genome browser. Wherever there are replicates, only the first one is selected. All data sets are in BED format. The score column is extracted for plotting and annotation.</p>
			</div>
		    </div>
		</div>
		<div class="panel panel-default">
		    <div class="panel-heading">
			<h4 class="panel-title">
			    <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion" href="#q3">Where does the interaction data come from? How is it compiled?</a>
			</h4>
		    </div>
		    <div id="q3" class="panel-collapse collapse">
			<div class="panel-body">
			    <p>The interaction matrix data comes from the data published in the original HiC method paper: <a href="http://www.sciencemag.org/content/326/5950/289">Liberman-Aiden et al, Science, 2009</a>. The resolution is 100 Kbp for intrachromosomal interaction and 1 Mbp for interchromosomal interaction. The ratio of Observed/Expected is reported for each region in the heatmap plot. The original genome build is hg18, UCSC liftover tool was applied to convert it to hg19.</p>
			</div>
		    </div>
		</div>
		<div class="panel panel-default">
		    <div class="panel-heading">
			<h4 class="panel-title">
			    <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion" href="#q4">How to generate multiple plots quickly?</a>
			</h4>
		    </div>
		    <div id="q4" class="panel-collapse collapse">
			<div class="panel-body">
	    <p>If you want to plot over multiple regions, just open the dialogue for multiple region specification on the submission page. If you want to try different combinations of, say, ENCODE data tracks, you can simply change your settings and submit multiple requests.</p>
			</div>
		    </div>
		</div>
		<div class="panel panel-default">
		    <div class="panel-heading">
			<h4 class="panel-title">
			    <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion" href="#q5">What are the files with suffix '.invalid_input'?</a>
			</h4>
		    </div>
		    <div id="q5" class="panel-collapse collapse">
			<div class="panel-body">
	    <p>The query file uploaded by user will first be annotated by ANNOVAR. If any record is not accepted by ANNOVAR, it will go into a file ending with '.invalid_input'. To understand what is the standard ANNOVAR format, please go <a href='http://www.openbioinformatics.org/annovar/annovar_input.html#standard'>here</a>.</p>
			</div>
		    </div>
		</div>
		<div class="panel panel-default">
		    <div class="panel-heading">
			<h4 class="panel-title">
			    <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion" href="#q6">What does score mean in the UChicago eQTL plot?</a>
			</h4>
		    </div>
		    <div id="q6" class="panel-collapse collapse">
			<div class="panel-body">
			    <p>The score is potentially different for each study. Mostly it is a -log10 P-value, but sometimes it can be a posterior probability or log10 Bayes Factor.  It really depends on what study is reported for that eQTL. Enlight does not differentiate among different studies used by UChicago eQTL browser. Please check out the text annotation and look at <a href='http://eqtl.uchicago.edu/cgi-bin/gbrowse/eqtl/?help=citations#Veyrieras_Pvalue'>here</a> to figure out which study is reported and what the score means.</p>
			</div>
		    </div>
		</div>
		<div class="panel panel-default">
		    <div class="panel-heading">
			<h4 class="panel-title">
			    <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion" href="#q7">What if multiple p values or scores are available for the same locus?</a>
			</h4>
		    </div>
		    <div id="q7" class="panel-collapse collapse">
			<div class="panel-body">
	    <p>Under such circumstances, the highest score (smallest p value) will be plotted.</p>
			</div>
		    </div>
		</div>
		<div class="panel panel-default">
		    <div class="panel-heading">
			<h4 class="panel-title">
			    <a class="accordion-toggle" data-toggle="collapse" data-parent="#accordion" href="#q8">Why no plot generated? OR why is the number of plots generated smaller than number of regions specified?</a>
			</h4>
		    </div>
		    <div id="q8" class="panel-collapse collapse">
			<div class="panel-body">
	    <p>Most likely it is because there is no SNP in the region you specified. If you believe there should be more figures generated, please email me (yunfeigu@usc.edu) detailed information about your submission (input file, region specification, other settings).</p>
			</div>
		    </div>
		</div>
	</div><!--panel-group-->
	<hr>
	<footer>
	<p>&copy; Wang Genomics Lab 2013-2014</p>
	</footer>
    </div> <!-- /container -->


    <!-- Bootstrap core JavaScript
    ================================================== -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
    <script src="../js/bootstrap.min.js"></script>
</body>
</html>
