<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<title>Enlight</title>
<meta name="keywords" content="green jelly, theme, free templates, nivo image slider, website, templatemo, CSS, HTML" />
<meta name="description" content="Green Jelly Theme, free CSS template provided by templatemo.com" />
<link href="../css/templatemo_style.css" rel="stylesheet" type="text/css" />

<link rel="stylesheet" href="../css/nivo-slider.css" type="text/css" media="screen" />

<script language="javascript" type="text/javascript">
function clearText(field)
{
    if (field.defaultValue == field.value) field.value = '';
    else if (field.value == '') field.value = field.defaultValue;
}
</script>

<link rel="stylesheet" type="text/css" href="../css/ddsmoothmenu.css" />

<script src="http://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
<script type="text/javascript" src="../js/ddsmoothmenu.js">

/***********************************************
* Smooth Navigational Menu- (c) Dynamic Drive DHTML code library (www.dynamicdrive.com)
* This notice MUST stay intact for legal use
* Visit Dynamic Drive at http://www.dynamicdrive.com/ for full source code
***********************************************/

</script>

<script type="text/javascript">

ddsmoothmenu.init({
	mainmenuid: "templatemo_menu", //menu DIV id
	orientation: 'h', //Horizontal or vertical menu: Set to "h" or "v"
	classname: 'ddsmoothmenu', //class added to menu's outer DIV
	//customtheme: ["#1c5a80", "#18374a"],
	contentsource: "markup" //"markup" or ["container_id", "path_to_menu_file"]
})

</script>
  
</head>
<body>

<div id="templatemo_wrapper">
	<div id="templatemo_header"></div> <!-- end of header -->
      <div id="templatemo_menu" class="ddsmoothmenu">
        <ul>
            <li><a href="../index.html" >Home</a></li>
            <li><a href="tutorial.html" >Tutorial</a></li>
            <li><a href="help.html">Help</a></li>
            <li><a href="pipeline.html">Pipeline</a> </li>
            <li><a href="faq.html" class="selected">FAQ</a></li>
	    <li><a href="#">Example</a>
	    <ul>
		<li><a href="example/exampleinput.html">Example Input</a></li>
		<li><a href="example/exampleoutput/index.html">Example Output</a></li>
	    </ul>
	    </li>
            </ul>
        <br style="clear: left" />
    </div> <!-- end of menu -->
    
    <script type="text/javascript" src="../js/jquery.nivo.slider.js"></script>
    <script type="text/javascript">
    $(window).load(function() {
    $('#slider').nivoSlider();
    });
    </script>
    
    <div id="templatemo_main"><br class="cleaner" />
	<h2>Enlight FAQ</h2>
	<div id="table-of-content"></div>

	<h3 id='q1'>Where does the eQTL data come from? How is it compiled?</h3>
	<p>The eQTL data sets come from two sources. The first one is the <a href="http://eqtl.uchicago.edu/cgi-bin/gbrowse/eqtl/">eQTL browser</a> of Pritchard Lab at University of Chicago, the other one is the <a href="http://www.ncbi.nlm.nih.gov/projects/gap/eqtl/index.cgi">GTEx project</a>. The GTEx eQTL data is split into different subsets based on analysis IDs. Analysis IDs usually represent different tissues. However, sometimes they merely represent different measurements for expression. The name of a data set can tell you the tissue and analysis assay (if more than one are assays are used). Tissue-specific information is unavailable for data from University of Chicago as no such meta-data is provided.</p>
	<br>
	<h3 id='q2'>Where does the ENCODE data come from? How is it compiled?</h3>
	<p>All ENCODE data sets come from UCSC genome browser. Wherever there are replicates, only the first one is selected. All data sets are in BED format. The score column is extracted for plotting and annotation.</p>
	<br>
	<h3 id='q3'>Where does the interaction data come from? How is it compiled?</h3>
	<p>The interaction matrix data comes from the data published in the original HiC method paper: <a href="http://www.sciencemag.org/content/326/5950/289">Liberman-Aiden et al, Science, 2009</a>. The resolution is 100 Kbp for intrachromosomal interaction and 1 Mbp for interchromosomal interaction. The ratio of Observed/Expected is reported for each region in the heatmap plot. The original genome build is hg18, UCSC liftover tool was applied to convert it to hg19.
	</p>
	<br>
	<h3 id='q4'>How to generate multiple plots quickly</h3>
	<p>If you want to plot over multiple regions, just open the dialogue for multiple region specification on the submission page. If you want to try different combinations of, say, ENCODE data tracks, you can simply change your settings and submit multiple requests.</p>
	<br>
	<h3 id='q5'>What are the files with suffix '.invalid_input'?</h3>
	<p>The query file uploaded by user will first be annotated by ANNOVAR. If any record is not accepted by ANNOVAR, it will go into a file ending with '.invalid_input'. To understand what is the standard ANNOVAR format, please go <a href='http://www.openbioinformatics.org/annovar/annovar_input.html#standard'>here</a>.</p>
	<br>
	<h3 id='q6'>What does score mean in the UChicago eQTL plot?</h3>
	<p>The score is potentially different for each study. Mostly it is a -log10 P-value, but sometimes it can be a posterior probability or log10 Bayes Factor.  It really depends on what study is reported for that eQTL. Enlight does not differentiate among different studies used by UChicago eQTL browser. Please check out the text annotation and look at <a href='http://eqtl.uchicago.edu/cgi-bin/gbrowse/eqtl/?help=citations#Veyrieras_Pvalue'>here</a> to figure out which study is reported and what the score means.
	</p>
	<br>
	<h3 id='q7'>What if multiple p values or scores are available for the same locus?</h3>
	<p>Under such circumstances, the highest score (smallest p value) will be plotted.</p>
	<br>

    </div> <!-- end of main -->
    
    <div id="templatemo_footer">
      <div class="cleaner"></div>
    </div> <!-- end of footer -->
</div>

<div id="templatemo_cr_bar_wrapper">
	<div id="templatemo_cr_bar">
    	Copyright 2014 <a href="http://genomics.usc.edu">USC Wang Lab</a> | Designed by <a href="http://www.templatemo.com" target="_parent">Free CSS Templates</a>
    </div>
</div>


<script type="text/javascript" src="../js/custom.js">
<script type='text/javascript' src='../js/logging.js'></script>
</body>
</html>
