<!DOCTYPE html>
<html lang="en">
    <head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<meta name="description" content="">
	<meta name="author" content="">
	<link rel="icon" href="images/favicon.ico">

	<title>Enlight</title>

	<!-- Bootstrap core CSS -->
	<link href="../css/bootstrap.min.css" rel="stylesheet">

	<!-- Custom styles for this template -->
	<link href="../css/jumbotron.css" rel="stylesheet">

	<!-- IE10 viewport hack for Surface/desktop Windows 8 bug -->
	<script src="../js/ie10-viewport-bug-workaround.js"></script>

	<!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
	<!--[if lt IE 9]>
	<script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
	<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
	<![endif]-->

	<script>
	    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
		    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
		m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
	    
	    ga('create', 'UA-50646209-1', 'usc.edu');
	    ga('send', 'pageview');
	    
	</script>

    </head>

    <body>

	<div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
	    <div class="container">
		<div class="navbar-header">
		    <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
			<span class="sr-only">Toggle navigation</span>
			<span class="icon-bar"></span>
			<span class="icon-bar"></span>
			<span class="icon-bar"></span>
		    </button>
		    <a class="navbar-brand" href="../index.html">Enlight</a>
		</div>
		<div class="navbar-collapse collapse">
		    <ul class="nav navbar-nav">
			<li ><a href="../index.html">Home</a></li>
			<li><a href="tutorial.html">Tutorial</a></li>
			<li class="active"><a href="help.html">Help</a></li>
			<li><a href="pipeline.html">Pipeline</a></li>
			<li><a href="faq.html">FAQ</a></li>
			<li class="dropdown">
			<a href="#" class="dropdown-toggle" data-toggle="dropdown">Example <span class="caret"></span></a>
			<ul class="dropdown-menu" role="menu">
			    <li><a href="example/exampleinput.html">Example Input</a></li>
			    <li><a href="example/exampleoutput/index.html">Example Output</a></li>
			</ul>
			</li>
		    </ul>
		</div><!--/.nav-collapse -->
	    </div>
	</div>

	<div class="container">
	<h2 id="exampleinput">Example Input</h2>
	<img width=600 height=340 src="../images/enlight_example_input.png" class='center'>
	<h2 id="exampleoutput_plot">Example Output (plot)</h2>
	<p><b>An example plot showing 1MB and 100KB region around rs10318, which is a known variant showing genome-wide significant association with colorectal cancer. After zooming in, DNase Hypersensitivity Sites (DHS) can be visually identified near some variants that are in high LD with the index SNP. These SNPs might be good candidates for causal variants. Plot summary and legend (not shown here) can be found on the second page of your result PDF file. The rest of the pages are summary plots showing number of variants in each LD interval, chromHMM category, or the variant count within different LD interval for each annotation dat track.</b></p>
	<br />
	<img width=900 height=1400 src="../images/enlight_example_output.png" class='center'>
	<img width=900 height=700 src="../images/enlight_example_output_summary.png" class='center'>
	<h2 id="exampleoutput_text">Example Text Annotation</h2>
	<p><b>If you opt to output ANNOVAR annotation, you will get a '*.hg19_multianno.html' on result page. This is the text annotation for each variant in the input file. ANNOVAR will output gene annotation, minor allele frequency (MAF) from 1000 Genomes project as well as selected UCSC datatracks and other annotation info (eg custome BED file, eQTL etc). If your input file does not have the 5 columns required by ANNOVAR (chr,start,end,alt,ref), Enlight will first convert rsIDs to chromosomal positions according to dbSNP137, and then run annotation.  Sometimes, you will see *.invalid_input files on result page, they contain lines that can't by recognized by ANNOVAR, e.g. lines that don't have chr/start/end.</b></p>
	<br />
	<img  width=900 height=300 src="../images/enlight_example_text_output.png" class='center'>
	<p>NOTE: The above output has been adjusted for display purposes. Actual results will differ.</p>
	<br /><br />
	</div><!--panel-group-->
	<hr>
	<footer>
	<p>&copy; Wang Genomics Lab 2013-2014</p>
	</footer>
    </div> <!-- /container -->


    <!-- Bootstrap core JavaScript
    ================================================== -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
    <script src="../js/bootstrap.min.js"></script>
</body>
</html>
